---
title: "CCLE_LUSC"
author: "AL"
date: "`r Sys.Date()`"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#Load libraries and setwd
library(here)
library(readxl)
library(stringr)
#library(ConsensusClusterPlus)
library(ComplexHeatmap)
library(circlize)
library(enrichR)
#library(immunedeconv)
library(ggplot2)
#library(iClusterPlus)
library(dplyr)
library(tidyr)
library(reshape2)
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 60), tidy = TRUE)
set.seed(2022)

getwd()
#setwd("smb://hlm/data/project/LungCancerMetabolismProgram")
```

## Load files
Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.

```{r load files, echo = FALSE}
#RNAseq data
CCLEtranscript = read.csv("~/Documents/CCLE_LUSC/Resources/CCLE_RNAseq_transcripts.csv", stringsAsFactors = FALSE)
head(CCLEtranscript)


#p63 transcripts matched to ENSEMBL IDs
TP63transcriptsisoform_identifiers = read.csv("~/Documents/CCLE_LUSC/Resources/Transcripts_p63Ntermtype.csv", stringsAsFactors = FALSE)
TP63transcriptsisoform_identifiers


#sample info
sample_meta = read.csv("~/Documents/CCLE_LUSC/Resources/sample_info.csv", stringsAsFactors = FALSE)
dim(sample_meta)
sample_meta


#identify specific cells that matched
cellsOfInterest = read.csv("~/Documents/CCLE_LUSC/Resources/sample_info_NSCLC.csv", stringsAsFactors = FALSE, header = TRUE)
head(cellsOfInterest)
dim(cellsOfInterest) #60 x 29
colnames(cellsOfInterest)

```

```{r organize df, echo=FALSE}

#pull out only TP63 transcripts
CCLEtranscript[1:5,1:5]
      
CCLEcolumn = colnames(CCLEtranscript) 
#every column name is a geneID
CCLEcolumnNumbers_forTP63 = grep("TP63", CCLEcolumn)
CCLEcolumnNumbers_forTP63
length(CCLEcolumnNumbers_forTP63) #14

#add depmap # to columns wanted
CCLEcolumnNumbers_forTP63
CCLEcolumnNumbers_forTP63 <- c(1, CCLEcolumnNumbers_forTP63)
length(CCLEcolumnNumbers_forTP63) #15
#add column 1 for depmap id#


#separate into TA v deltaN vectors
TAp63_isoforms = TP63transcriptsisoform_identifiers$Nterm %in% "TAp63"
table(TAp63_isoforms) #8 false and 6 true
TAp63_isoforms = subset(TP63transcriptsisoform_identifiers, TP63transcriptsisoform_identifiers$Nterm %in% "TAp63")
TAp63_isoforms #note the Transcript.ID column starts with ENST and doesn't have any periods 
dim(TAp63_isoforms) # 6 x 9
TAp63_isoforms = TAp63_isoforms[,1]
TAp63_isoforms

deltaNp63_isoforms = subset(TP63transcriptsisoform_identifiers, TP63transcriptsisoform_identifiers$Nterm %in% "DNp63")
deltaNp63_isoforms
dim(deltaNp63_isoforms) #8 x9
deltaNp63_isoforms = deltaNp63_isoforms[,1]
deltaNp63_isoforms


#pull out sample_meta file make mini df with depmapID and strippedcellname only
sample_meta_names = sample_meta[,c(1,3)]
sample_meta_names
dim(sample_meta_names) #1804 x 2


#replace CCLEtranscript depmap id to stripped cell line name for downstream labeling
CCLEtranscript_CellsRelabel = CCLEtranscript

compare = table(CCLEtranscript_CellsRelabel$X %in% sample_meta$DepMap_ID)
compare #1406 true
compare = table(sample_meta$DepMap_ID %in% CCLEtranscript_CellsRelabel$X)
compare #434 false and 1406 true

#rename depmapID with stripped cell name 
CCLEtranscript_CellsRelabel[1:5,1:5]
names(CCLEtranscript_CellsRelabel)[1] = "DepMap_ID"
CCLEtranscript_CellsRelabel = left_join(CCLEtranscript_CellsRelabel, sample_meta_names, by = join_by(DepMap_ID))
dim(CCLEtranscript_CellsRelabel) #1406 x 227664
CCLEtranscript_CellsRelabel[1:5,227662:227664]
CCLEtranscript_CellsRelabel = CCLEtranscript_CellsRelabel %>%
    relocate(stripped_cell_line_name, .after = DepMap_ID)
head(CCLEtranscript_CellsRelabel)
```

```{r subset lung cancer and TP63, echo=FALSE}
CCLEtranscript_CellsRelabel[1:5, 1:5] #has depmadID then stripped cell name
CCLEtranscript[1:5,1:5] #is by depmap id# under colname X
CCLEtranscript_SCN = CCLEtranscript_CellsRelabel[,-1] #is just stripped cell name
CCLEtranscript_SCN[1:5, 1:5]

MatchedCells = cellsOfInterest$DepMap_ID %in% CCLEtranscript$X
table(MatchedCells) #5 false 55 true

#Move forward with only matched cells - note making a new table using subset function but with all gene transcripts from original
MatchedCells = subset(CCLEtranscript, CCLEtranscript_SCN$stripped_cell_line_name %in% cellsOfInterest$stripped_cell_line_name)
dim(MatchedCells) #55 x 227663


#make new table with only interested rows (cell lines) and columns (TP63 transcripts)
CCLEtable = MatchedCells[,CCLEcolumnNumbers_forTP63]
head(CCLEtable) #DepMap ID first column under X
dim(CCLEtable) #55x 15
CCLEtable #col names start with TP63..ENST then end with a period, need to edit to match
CCLEtable_colnames = colnames(CCLEtable) %>%
    str_sub(start = 7, end = -2)


CCLEtable_colnames #accidentally removed the first colname need to rename to DepMap_ID
colnames(CCLEtable) = CCLEtable_colnames
colnames(CCLEtable)[1] = "DepMap_ID"
head(CCLEtable)

TP63transcriptsisoform_identifiers #use column Transcript.ID rows looks to match CCLETable names
compare = TP63transcriptsisoform_identifiers$Transcript.ID %in% colnames(CCLEtable)
table(compare) #all 14 TRUE

```

```{r pressure, echo=FALSE}

head(CCLEtable) #DepMap_ID is column 1

deltaNp63_isoforms
TAp63_isoforms

sortorder = c("DepMap_ID",TAp63_isoforms,deltaNp63_isoforms)
sortorder


CCLEtable_isoforms = CCLEtable[,sortorder]
CCLEtable_isoforms
dim(CCLEtable_isoforms)
#55 x 15 in correct order note that TAp63 are columns 2:7, and deltaNp63 are 8:15

TAp63_sum = rowSums(CCLEtable_isoforms[,c(2:7)], na.rm=TRUE)
deltaNp63_sum = rowSums(CCLEtable_isoforms[,c(8:15)], na.rm = TRUE)

CCLEtable_isoforms_avgs = data.frame(CCLEtable_isoforms[,1],
                                          TAp63_sum,
                                         deltaNp63_sum )
colnames(CCLEtable_isoforms_avgs)[1]="DepMap_ID"
CCLEtable_isoforms_avgs #need to add lung cancer type column 
#use cellsOfInterest$lineage_sub_subtype
cellsOfInterest
cellsOfInterest_type = cellsOfInterest[,c("DepMap_ID", "stripped_cell_line_name", "lineage_sub_subtype")]
cellsOfInterest_type
cellsOfInterest_type = left_join(cellsOfInterest_type,CCLEtable_isoforms_avgs)
cellsOfInterest_type #file containing cell info and lineage type and sums of TAp63 and DNp63
write.csv(cellsOfInterest_type,file="~/Documents/CCLE_LUSC/Exports/EXPORT_Lung_p63isoformsCCLEdata.csv")

cellsOfInterest_type$lineage_sub_subtype == 'NSCLC_adenocarcinoma'
p63isoforms_LUAD = subset(cellsOfInterest_type, cellsOfInterest_type$lineage_sub_subtype == 'NSCLC_adenocarcinoma')
p63isoforms_LUAD

cellsOfInterest_type$lineage_sub_subtype == 'NSCLC_squamous'
p63isoforms_LUSC = subset(cellsOfInterest_type, cellsOfInterest_type$lineage_sub_subtype == 'NSCLC_squamous')
p63isoforms_LUSC


cellsOfInterest_type$lineage_sub_subtype == 'NSCLC_adenosquamous'
p63isoforms_LuAdSq = subset(cellsOfInterest_type, cellsOfInterest_type$lineage_sub_subtype == 'NSCLC_adenosquamous')
p63isoforms_LuAdSq



#make barplots for each TAp63 isoform
barplot(p63isoforms_LUSC$TAp63_sum, names = p63isoforms_LUSC$stripped_cell_line_name, las = 2, main = "CCLE LUAD Cell lines - TAp63 Expression", ylab = "summed expected counts")

barplot(p63isoforms_LUAD$TAp63_sum, names = p63isoforms_LUAD$stripped_cell_line_name, las = 2, main = "CCLE LUAD Cell lines - TAp63 Expression", ylab = "summed expected counts")
#make font smaller because labeling out of spacing

#make barplots for each DNp63 isoform
barplot(p63isoforms_LUSC$deltaNp63_sum, names = p63isoforms_LUSC$stripped_cell_line_name, las = 2, main = "CCLE LUAD Cell lines - DNp63 Expression", ylab = "summed expected counts")


```

```{r pressure, echo=FALSE}
#NEED TO EDIT TITLES

#create a stacked bar chart
colnames(cellsOfInterest_type)
cellsOfInterest_type_longer = pivot_longer(cellsOfInterest_type, cols = c('TAp63_sum','deltaNp63_sum'), names_to = 'p63_isoform', values_to = 'sums')
cellsOfInterest_type_longer

p63isoforms_LUSC_longer = pivot_longer(p63isoforms_LUSC, cols = c('TAp63_sum','deltaNp63_sum'), names_to = 'p63_isoform', values_to = 'sums')
p63isoforms_LUSC_longer
p63isoforms_LUSC_longer = p63isoforms_LUSC_longer[,c(2,4,5)]
p63isoforms_LUSC_longer

stacked_summed = ggplot(p63isoforms_LUSC_longer, aes(fill=p63_isoform, y=sums, x=stripped_cell_line_name)) +
    geom_bar(position="stack", stat="identity") + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
stacked_summed


data_percentage = p63isoforms_LUSC_longer
data_percentage = data_percentage %>%
    group_by(stripped_cell_line_name) %>%
    mutate(percent = 100*sums/sum(sums))
data_percentage
colnames(data_percentage) = c("Cell_Line", "TP63_isoform", "Summed_Counts", "percent") 
#concatenated another column for percentages

stacked = ggplot(data_percentage, aes(x = Cell_Line, y = percent, fill = TP63_isoform)) +
  geom_bar(position = "fill", stat = "identity") + 
  scale_y_continuous(labels = scales::percent_format()) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
stacked
#stacked bar plots per cell type



```

# Wilkerson Classifiers
```{r pressure, echo=FALSE}
cellsOfInterest
#currently looking at LUAD cells of interest
length(cellsOfInterest)

#Wilkerson subtype
#download expression full  RNAseq data genes for gene level data instead of transcript variants we were using for analysis above

#GENE LEVEL DATA (TPM gene expression data for all genes using RSEM. Log2 transformed, using a pseudo-count of 1)
#Rows: cell lines (Broad IDs) and Columns: genes (HGNC symbol and Ensembl ID)
CCLE_ExpFull_RNAseq = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/CCLE_expression_full.csv", stringsAsFactors = FALSE, check.names=FALSE, header=TRUE, row.names=1)
CCLE_ExpFull_RNAseq[1:5,1:5]
dim(CCLE_ExpFull_RNAseq)
#  1406 19222 and genes are columns and depmap ID/cells are rows under column name X


compare = cellsOfInterest %in% row.names(CCLE_ExpFull_RNAseq)
table(compare)
#false 3 and true 38

MatchedCells = cellsOfInterest %in% row.names(CCLE_ExpFull_RNAseq)
MatchedCells
table(MatchedCells)
#confirmed 3 don't match, move forward with only matched cells
CCLE_ExpFull_RNAseq_subset = subset(CCLE_ExpFull_RNAseq, row.names(CCLE_ExpFull_RNAseq) %in% cellsOfInterest)
dim(CCLE_ExpFull_RNAseq_subset)
#see 38 rows - okay, but still 19222 columns need to trim to wilkerson gene set


#upload gene signatures from wilkerson files
wilkerson_centroid = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/wilkersonLUSCsignatures/centroids_all_cohorts_all_common_genes.csv", row.names = 1)
head(wilkerson_centroid)
Wilkerson_genes = row.names(wilkerson_centroid)
Wilkerson_genes
length(Wilkerson_genes)

#get genes in CCLE_ExpFull_RNAseq
CCLE_colnames = colnames(CCLE_ExpFull_RNAseq)
length(CCLE_colnames)
CCLE_colnames[1:5]
#removed all characters after . in names
CCLE_colnames = gsub(" .*","",CCLE_colnames)
CCLE_colnames[1:5]

#reapply colnames
CCLE_ExpFull_RNAseq_subset_NoNums=CCLE_ExpFull_RNAseq_subset
colnames(CCLE_ExpFull_RNAseq_subset_NoNums) = CCLE_colnames
head(CCLE_ExpFull_RNAseq_subset_NoNums)



compare = Wilkerson_genes %in% CCLE_colnames
head(compare)
table(compare)
compare_genes = Wilkerson_genes[compare]

compare2 = CCLE_colnames %in%  Wilkerson_genes
table(compare2)

#genes # matched 8421, checked genes on HUGO and saw some gene names are outdated
Wilkerson_not_matching_genes = Wilkerson_genes[!compare]
length(Wilkerson_not_matching_genes) #940 matches check before
#CCLE_colnames[grep("^KRT.*", CCLE_colnames, ignore.case = FALSE)]


CCLE_ExpFull_RNAseq_subset_NoNums[1:5,1:5]
CCLE_ExpFull_RNAseq_subset_NoNums[,1]
dim(CCLE_ExpFull_RNAseq_subset_NoNums)

#cannot do - transpose first
CCLE_Wilkerson <- CCLE_ExpFull_RNAseq_subset_NoNums[,compare_genes]
dim(CCLE_Wilkerson)
CCLE_Wilkerson[1:5,1:5]

t_CCLE_Wilkerson = t(CCLE_Wilkerson)
t_CCLE_Wilkerson[1:5,1:5]
dim(t_CCLE_Wilkerson)
#subset -- genes are rows and samples are columns

```



#Generate long df of Mapping_table from Eric Welsh with symbols and alias for use
```{r}
#read in excel table for gene matching from Eric Welsh
Mapping_table = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/genemapping/gencode_v19_merged_gene_transcript_annotation.csv", stringsAsFactors = FALSE, header = TRUE)
head(Mapping_table)


#pull out symbols and alias column from mapping_table
head(Mapping_table)
Mapping_table_symbol_alias = select(Mapping_table, symbol, alias)
head(Mapping_table_symbol_alias)

#remove all rows with empty values in alias
Mapping_table_symbol_alias = Mapping_table_symbol_alias[!(Mapping_table_symbol_alias$alias == ""),]
head(Mapping_table_symbol_alias)
dim(Mapping_table_symbol_alias)

#what is the total amount of new columns you will need (how many spaces are there in the largest alias plus 1 for the extra alias)
ncols <- max(stringr::str_count(Mapping_table_symbol_alias$alias, " ")) + 1
ncols #31 columns after splitting

#make column names
colmn = paste ("alias", 1:ncols)
colmn


#split multiple names under alias separated by space
Mapping_table_symbol_alias = separate(Mapping_table_symbol_alias, col = alias, sep = " ", into = colmn)
head(Mapping_table_symbol_alias)

#make into a long table using gather from tidyr
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias %>%
  gather(all_of(colmn), key = 'amount', value ='Alias')

head(Mapping_table_symbol_alias_long)
Mapping_table_symbol_alias_long
dim(Mapping_table_symbol_alias_long) #696322 x 3

#check if NA's were included, check to see if other aliases were included because only see alias1
length(which(Mapping_table_symbol_alias_long$amount=="alias 10")) 
length(which(Mapping_table_symbol_alias_long$amount=="alias 30")) 
#for every alias# there is 22462 guessing all the NAs were included need to remove
length(which(Mapping_table_symbol_alias_long$amount=="NA")) #this says 0, may be it is blank?
#didn't work    length(which(Mapping_table_symbol_alias_long$amount==is.na)) 
table(sapply(Mapping_table_symbol_alias_long, is.na)) #643286 is TRUE, 1445680 is false
#remove empty cells
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias_long[-which(is.na(Mapping_table_symbol_alias_long$Alias)),]
dim(Mapping_table_symbol_alias_long) #53036 x 3

#order by gene name
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias_long[order(Mapping_table_symbol_alias_long$symbol),]
head(Mapping_table_symbol_alias_long)
length(which(Mapping_table_symbol_alias_long$amount=="alias 10")) #74
length(which(Mapping_table_symbol_alias_long$amount=="alias 30")) #1

write.csv(Mapping_table_symbol_alias_long, file="~/Documents/Rstudio/LUSC/Gene_Mapping_Alias_Long.csv")


# don't want to know how many alias there is per gene can remove column 2
Mapping_table_symbol_alias_long = select(Mapping_table_symbol_alias_long, -amount)
head(Mapping_table_symbol_alias_long)
#finally have mapping table with column 1 being the symbol and column two being the alias

```

#clean up genes comparisons
```{r}
head(Wilkerson_not_matching_genes)

compare = select(Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$symbol)
table(compare) #579 FALSE, and 361 TRUE


compare = Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$Alias
table(compare) #382 FALSE, and 558 TRUE


compare1 = subset(Wilkerson_not_matching_genes, Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$symbol)
length(compare1) #361

compare2 = subset(Wilkerson_not_matching_genes, Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$Alias)
length(compare2) #558


compare_both = compare1 %in% compare2
table(compare_both) #352 FALSE, 9 TRUE

compare_both = subset(compare1, compare1 %in% compare2)
compare_both


compare_both = compare2 %in% compare1
table(compare_both) #549 FALSE, 9 TRUE

compare_both = subset(Wilkerson_not_matching_genes, compare1 %in% compare2)
compare_both


#unique from symbols
unique_compare1 = subset(compare1, !(compare1 %in% compare2))
unique_compare1
#combine for a total covered list
pot_target_gene_names = c(unique_compare1, compare2)
length(pot_target_gene_names) #910
#check for duplicates
table(duplicated(pot_target_gene_names)) #910 FALSE

unmatched_wilkerson = subset(Wilkerson_not_matching_genes, !(Wilkerson_not_matching_genes %in% pot_target_gene_names))
length(unmatched_wilkerson) #30


#need to change names for Wilkerson_not_matching genes 

#also need to check gene names for RNAseq 


```


#Change Wilkerson alias to symbol names
```{r}
head(wilkerson_centroid)
colnames(wilkerson_centroid)
dim(wilkerson_centroid)
#issue that column 1 has the gene names and don't know how to target it - row names is the gene alias - read in data again and not put row.names = TRUE
wilkerson_centroid_mappedgenes = wilkerson_centroid

head(wilkerson_centroid_mappedgenes)
#row.names(wilkerson_centroid_mappedgenes)[1] = "Ashley" ---this changed the first row to name Ashley
  

#first check to see if wilkerson gene centroid names match CCLE RNAseq, 
#if not then check alias table and see if alias match and change to gene symbol name
head(Mapping_table_symbol_alias_long) #alias table
head(CCLE_colnames) #gene names from the CCLE RNAseq 

for (i in 1:nrow(wilkerson_centroid_mappedgenes)) {
  print(i)
  current_gene = row.names(wilkerson_centroid_mappedgenes)[i]
  # need if statement here; does the current_gene match CCLE?
  if (!current_gene %in% CCLE_colnames) {
    # does not match, need to get aliases for this gene symbol
    if (current_gene %in% Mapping_table_symbol_alias_long$Alias) {
      # gets the alias to gene symbol mapping
      current_gene_aliases = subset(Mapping_table_symbol_alias_long, Alias == current_gene)
      for (j in 1:nrow(current_gene_aliases)) {
        # check if the mapping/new gene symbol is already in Wilkerson
        if (!current_gene_aliases$symbol[j] %in% row.names(CCLE_colnames)) {
        # assigns the mapping back into the wilkerson_centroid_mappedgenes data frame
        row.names(wilkerson_centroid_mappedgenes)[i] = current_gene_aliases$symbol
      }
      }
    }
  }
}

```


#Try For Loop
```{r}

#Change Wilkerson alias to symbol names
```{r}
head(wilkerson_centroid)
colnames(wilkerson_centroid)
dim(wilkerson_centroid)
#issue that column 1 has the gene names and don't know how to target it - row names is the gene alias - read in data again and not put row.names = TRUE
wilkerson_centroid_mappedgenes = wilkerson_centroid

head(wilkerson_centroid_mappedgenes)
#row.names(wilkerson_centroid_mappedgenes)[1] = "Ashley" ---this changed the first row to name Ashley
  

#first check to see if wilkerson gene centroid names match CCLE RNAseq, 
#if not then check alias table and see if alias match and change to gene symbol name
head(Mapping_table_symbol_alias_long) #alias table
head(CCLE_colnames) #gene names from the CCLE RNAseq 

for (i in 1:nrow(wilkerson_centroid_mappedgenes)) {
  print(i)
  current_gene = row.names(wilkerson_centroid_mappedgenes)[i]
  # need if statement here; does the current_gene match CCLE?
  if (!current_gene %in% CCLE_colnames) {
    # does not match, need to get aliases for this gene symbol
    if (current_gene %in% Mapping_table_symbol_alias_long$Alias) {
      # gets the alias to gene symbol mapping
      current_gene_aliases = subset(Mapping_table_symbol_alias_long, Alias == current_gene)
      for (j in 1:nrow(current_gene_aliases)) {
        # check if the mapping/new gene symbol is already in Wilkerson
        if (!current_gene_aliases$symbol[j] %in% row.names(CCLE_colnames)) {
        # assigns the mapping back into the wilkerson_centroid_mappedgenes data frame
        row.names(wilkerson_centroid_mappedgenes)[i] = current_gene_aliases$symbol
      }
      }
    }
  }
}

```


#Try For Loop
```{r}
#Change Wilkerson alias to symbol names
```{r}
head(wilkerson_centroid)
colnames(wilkerson_centroid)
dim(wilkerson_centroid)
#issue that column 1 has the gene names and don't know how to target it - row names is the gene alias - read in data again and not put row.names = TRUE
wilkerson_centroid_mappedgenes = wilkerson_centroid

head(wilkerson_centroid_mappedgenes)
#row.names(wilkerson_centroid_mappedgenes)[1] = "Ashley" ---this changed the first row to name Ashley
  

#first check to see if wilkerson gene centroid names match CCLE RNAseq, 
#if not then check alias table and see if alias match and change to gene symbol name
head(Mapping_table_symbol_alias_long) #alias table
head(CCLE_colnames) #gene names from the CCLE RNAseq 

for (i in 1:nrow(wilkerson_centroid_mappedgenes)) {
  print(i)
  current_gene = row.names(wilkerson_centroid_mappedgenes)[i]
  # need if statement here; does the current_gene match CCLE?
  if (!current_gene %in% CCLE_colnames) {
    # does not match, need to get aliases for this gene symbol
    if (current_gene %in% Mapping_table_symbol_alias_long$Alias) {
      # gets the alias to gene symbol mapping
      current_gene_aliases = subset(Mapping_table_symbol_alias_long, Alias == current_gene)
      for (j in 1:nrow(current_gene_aliases)) {
        # check if the mapping/new gene symbol is already in Wilkerson
        if (!current_gene_aliases$symbol[j] %in% row.names(CCLE_colnames)) {
        # assigns the mapping back into the wilkerson_centroid_mappedgenes data frame
        row.names(wilkerson_centroid_mappedgenes)[i] = current_gene_aliases$symbol
      }
      }
    }
  }
}

```


#Try For Loop
```{r}

#try with selected genes - arbitrarily picked line 1
current_gene = Wilkerson_not_matching_genes


#only 53 matched, looking at alias column, different alias are separated by spaces--- write another for loop and it will be nested


#make sure you use length for this vector and nrow for df
#for(i in 1:length(Wilkerson_not_matching_genes)){
  print(Wilkerson_not_matching_genes[i])
  #if stop here then will only print the 940 gene names
  current_gene = Wilkerson_not_matching_genes[i]
  #every iteration new variable current_gene will be the gene of interest at the moment
  
  #current_gene is hopefully the alias and we want to change it to a symbol
#  }

  
  #Paul's method - one loop to scan every row, the see if there a space, then do nested loop to split string or use str_match function to see if
```

```

```