---
title: "CCLE_LUSC_Limma"
author: "AL"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
#Load libraries and setwd
library(here)
library(readxl)
library(stringr)
library(ConsensusClusterPlus)
library(ComplexHeatmap)
library(circlize)
library(enrichR)
library(immunedeconv)
library(ggplot2)
library(iClusterPlus)
library(dplyr)
library(tidyr)
library(reshape2)
library(limma)
library(edgeR)
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 60), tidy = TRUE)
set.seed(2022)

getwd()
```

#load files - LARGE RNAseq files - dont run unless necessary
```{r}
#transcript level data - RNAseq transcript tpm data using RSEM. Log2 transformed, using a pseudo-count of 1
CCLEtranscript = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/CCLE_RNAseq_transcripts.csv", stringsAsFactors = FALSE)
dim(CCLEtranscript) #1406 x 227663
head(CCLEtranscript)

#GENE LEVEL DATA (TPM gene expression data for all genes using RSEM. Log2 transformed, using a pseudo-count of 1)
#Rows: cell lines (Broad IDs) and Columns: genes (HGNC symbol and Ensembl ID)
#CCLE_ExpFull_RNAseq = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/CCLE_expression_full.csv", stringsAsFactors = FALSE, check.names=FALSE, header=TRUE, row.names=1)
#dim(CCLE_ExpFull_RNAseq) #1406 x 53970 


#RNAseq_reads (RSEM) Rows: cell lines (Broad IDs) and Columns: genes (HGNC symbol and Ensembl ID)
CCLE_RNAseq_reads = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/CCLE_RNAseq_reads.csv", header=TRUE, stringsAsFactors = FALSE, row.names = 1 )
dim(CCLE_RNAseq_reads) #1406 54357
head(CCLE_RNAseq_reads)


CCLEcelllines_classifiedbyWilkerson = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/LUSCclassifers_WilkersonTable2.csv",header=TRUE, stringsAsFactors = FALSE)
CCLEcelllines_classifiedbyWilkerson #28 x 9
```

#subgroup classifiers 
```{r}
colnames(CCLEcelllines_classifiedbyWilkerson)
LUSCcellsPredicted = CCLEcelllines_classifiedbyWilkerson[,c("Cell.line", "Predicted")]
colnames(LUSCcellsPredicted)[1] = "cell_line_name"
dim(LUSCcellsPredicted) #28 x 2
LUSCcellsPredicted

cellIDS = sample_meta[,1:3]
cellIDS

ClassicalLUSC = subset(LUSCcellsPredicted, Predicted == "Classical")
ClassicalLUSC = left_join(ClassicalLUSC, cellIDS)
ClassicalLUSC
ClassicalLUSCcells = ClassicalLUSC$stripped_cell_line_name
ClassicalLUSCcells #7



NotClassicalLUSC = subset(LUSCcellsPredicted, !LUSCcellsPredicted$Predicted == "Classical")
NotClassicalLUSC = left_join(NotClassicalLUSC, cellIDS)
NotClassicalLUSC
#unmatched: RERF-LC-Sq-1, HCC-1897


NotClassicalLUSC_unlabCells = grep("hcc", cellIDS$stripped_cell_line_name)
NotClassicalLUSC_unlabCells #none found
NotClassicalLUSC_unlabCells = grep("Sq", cellIDS$stripped_cell_line_name)
NotClassicalLUSC_unlabCells#none found

NotClassicalLUSC = na.omit(NotClassicalLUSC)
NotClassicalLUSCcells = NotClassicalLUSC$stripped_cell_line_name
NotClassicalLUSCcells #19 now since removed the two that didn't match

#cells of interest from RNASeq information
#LUSC = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/sample_info_LUSC.csv")
#dim(LUSC) #33 x 29
#head(LUSC)

#LUSC = LUSC %>%
  #left_join(LUSCcellsPredicted) %>%
  #relocate(Predicted, .after = CCLE_Name)
#dim(LUSC)
```

#organize sample_meta
```{r}
#sample info
sample_meta = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/sample_info.csv", stringsAsFactors = FALSE)
dim(sample_meta) #1840 x 29
colnames(sample_meta)
#instead of separating into two df add a meta file for group info

sample_meta
cellIDS
cellIDS = cellIDS [,-2]
cellIDS #1840 x 2


sample_meta = sample_meta %>%
  left_join(LUSCcellsPredicted) %>%
  relocate(Predicted, .after = CCLE_Name)
dim(sample_meta)
sample_meta
```


#add mutations to sample_meta
```{r}
CCLEmutations = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/CCLE_mutations.csv")
head(CCLEmutations)
colnames(CCLEmutations)
dim(CCLEmutations) #1235466 x 32 use columns called DepMap_ID, Hugo_Symbol, Protein_Change, 
length(unique(CCLEmutations$DepMap_ID)) #1771


CCLEmutations_subset_tp53nrf2keap1 = subset(CCLEmutations, Hugo_Symbol == "TP53" | Hugo_Symbol == "NFE2L2" | Hugo_Symbol == "KEAP1")
dim(CCLEmutations_subset_tp53nrf2keap1) #1513 by 32
colnames(CCLEmutations_subset_tp53nrf2keap1)
length(unique(CCLEmutations_subset_tp53nrf2keap1$DepMap_ID)) #1136

CCLEmutations_subset_proteinmutation = CCLEmutations_subset_tp53nrf2keap1[, c("DepMap_ID", "Hugo_Symbol", "Protein_Change")]
CCLEmutations_subset_proteinmutation = pivot_wider(CCLEmutations_subset_proteinmutation, names_from = "Hugo_Symbol", values_from = "Protein_Change")

sample_meta = left_join(sample_meta, CCLEmutations_subset_proteinmutation)
colnames(sample_meta)



```

#make CCLE dataframe
```{r}
RNASEQ = CCLE_RNAseq_reads #insert which RNAseq csv you want to use
head(RNASEQ)
dim(RNASEQ) #1406 x 54357

sample_meta_LUSC #35 x 34

cellsOfInterest = sample_meta_LUSC$DepMap_ID #make vector of cells of interest by Depmap ID#
cellsOfInterest

MatchedCells = cellsOfInterest %in% row.names(RNASEQ)
table(MatchedCells) #7 false and 28 true
notMatchedCells = !cellsOfInterest %in% row.names(RNASEQ)
notMatchedCells






cellsOfInterest[notMatchedCells] #[1] "ACH-001078" "ACH-001079" "ACH-001234" "ACH-000727" "ACH-001489" "ACH-002122" "ACH-002156"
cellIDS$DepMap_ID[notMatchedCells]

#Move forward with only matched cells - note making a new table using subset function but with all gene transcripts from original
MatchedCellsTable = subset(RNASEQ, row.names(RNASEQ) %in% cellsOfInterest)
head(MatchedCellsTable)
dim(MatchedCellsTable) #28x 54357

#get genes in CCLEtranscript from the MatchedCell subset
CCLEtranscript_colnames = colnames(MatchedCellsTable)
length(CCLEtranscript_colnames) #54357
CCLEtranscript_colnames[1:5] # have gene name then ..ENGS... need to remove the .. onwards
CCLEtranscript_colnames = str_replace(CCLEtranscript_colnames, "\\.\\..*", "")
colnames(MatchedCellsTable) = CCLEtranscript_colnames
head(MatchedCellsTable)
dim(MatchedCellsTable) #13 x 54357
#removed all characters after . in names
#names(RNASEQ) = str_replace(names(CCLE_RNAseq_reads), "\\.\\..*", "")
#\\. look at dot then do it twice then last . is for replacing with a wildcard then add * to repeat until the end


#currently has DepMap_IDs as colnames need to change to cell names
MatchedCellsTable_names = data.frame(rownames(MatchedCellsTable))
MatchedCellsTable_names
colnames(MatchedCellsTable_names)[1] = "DepMap_ID"
MatchedCellsTable_names = MatchedCellsTable_names %>%
  left_join(cellIDS)

MatchedCellsTable_names

rownames(MatchedCellsTable) = MatchedCellsTable_names$stripped_cell_line_name
MatchedCellsTable #28 x 54357
```

#generate Wilkerson subtypes using newer CCLE RNAseq data
```{r}

#upload gene signatures from wilkerson files
wilkerson_centroid = read.csv("~/Documents/CCLE_LUSC/Resources/centroids_all_cohorts_all_common_genes.csv", row.names = 1)
head(wilkerson_centroid)
Wilkerson_genes = row.names(wilkerson_centroid)
Wilkerson_genes
length(Wilkerson_genes) #9361


#get genes in CCLE_ExpFull_RNAseq
head(MatchedCellsTable)
CCLERNAseq_colnames = colnames(MatchedCellsTable)
length(CCLERNAseq_colnames) #54357
CCLERNAseq_colnames[1:5]


#Goal compare gene names from Wilkerson centroid to gene names from RNASeq
compare = Wilkerson_genes %in% CCLERNAseq_colnames
table(compare) #False 956 and True 8405 sooo 956 of the Wilkerson genes don't match what is in the CCLE RNASeq
compare_genes = Wilkerson_genes[compare]

compare2 = CCLERNAseq_colnames %in%  Wilkerson_genes
table(compare2) #false 45952 and true 8405

#genes # matched 8421, checked genes on HUGO and saw some gene names are outdated
Wilkerson_not_matching_genes = Wilkerson_genes[!compare]
length(Wilkerson_not_matching_genes) #956 matches check before
#CCLE_colnames[grep("^KRT.*", CCLE_colnames, ignore.case = FALSE)]

#needs to try to find a way to change the 956 unmatched genes - emailed Eric Welsh and tried to use his gene mapping table

```
#Generate long df of Mapping_table from Eric Welsh with symbols and alias for use
```{r}
#read in excel table for gene matching from Eric Welsh
Mapping_table = read.csv("~/Documents/CCLE_LUSC/Resources/gencode_v19_merged_gene_transcript_annotation.csv", stringsAsFactors = FALSE, header = TRUE)
head(Mapping_table)


#pull out symbols and alias column from mapping_table
head(Mapping_table)
Mapping_table_symbol_alias = select(Mapping_table, symbol, alias)
head(Mapping_table_symbol_alias)

#remove all rows with empty values in alias
Mapping_table_symbol_alias = Mapping_table_symbol_alias[!(Mapping_table_symbol_alias$alias == ""),]
head(Mapping_table_symbol_alias)
dim(Mapping_table_symbol_alias)

#what is the total amount of new columns you will need (how many spaces are there in the largest alias plus 1 for the extra alias)
ncols <- max(stringr::str_count(Mapping_table_symbol_alias$alias, " ")) + 1
ncols #31 columns after splitting

#make column names
colmn = paste ("alias", 1:ncols)
colmn


#split multiple names under alias separated by space
Mapping_table_symbol_alias = separate(Mapping_table_symbol_alias, col = alias, sep = " ", into = colmn)
head(Mapping_table_symbol_alias)

#make into a long table using gather from tidyr
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias %>%
  gather(all_of(colmn), key = 'amount', value ='Alias')

head(Mapping_table_symbol_alias_long)
Mapping_table_symbol_alias_long
dim(Mapping_table_symbol_alias_long) #696322 x 3

#check if NA's were included, check to see if other aliases were included because only see alias1
length(which(Mapping_table_symbol_alias_long$amount=="alias 10")) 
length(which(Mapping_table_symbol_alias_long$amount=="alias 30")) 
#for every alias# there is 22462 guessing all the NAs were included need to remove
length(which(Mapping_table_symbol_alias_long$amount=="NA")) #this says 0, may be it is blank?
#didn't work    length(which(Mapping_table_symbol_alias_long$amount==is.na)) 
table(sapply(Mapping_table_symbol_alias_long, is.na)) #643286 is TRUE, 1445680 is false
#remove empty cells
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias_long[-which(is.na(Mapping_table_symbol_alias_long$Alias)),]
dim(Mapping_table_symbol_alias_long) #53036 x 3

#order by gene name
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias_long[order(Mapping_table_symbol_alias_long$symbol),]
head(Mapping_table_symbol_alias_long)
length(which(Mapping_table_symbol_alias_long$amount=="alias 10")) #74
length(which(Mapping_table_symbol_alias_long$amount=="alias 30")) #1

write.csv(Mapping_table_symbol_alias_long, file="~/Documents/Rstudio/LUSC/Gene_Mapping_Alias_Long.csv")


# don't want to know how many alias there is per gene can remove column 2
Mapping_table_symbol_alias_long = select(Mapping_table_symbol_alias_long, -amount)
head(Mapping_table_symbol_alias_long)
#finally have mapping table with column 1 being the symbol and column two being the alias

```

#genes comparisons between wilkerson and RNAseq transcripts
```{r}
head(Wilkerson_not_matching_genes)

compare = Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$symbol
table(compare) #582 FALSE


compare = Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$Alias
table(compare) #398 FALSE, and 558 TRUE


compare1 = subset(Wilkerson_not_matching_genes, Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$symbol)
length(compare1) #374

compare2 = subset(Wilkerson_not_matching_genes, Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$Alias)
length(compare2) #558


compare_both = compare1 %in% compare2
table(compare_both) #365 FALSE, 9 TRUE

compare_both = subset(compare1, compare1 %in% compare2)
compare_both #"B3GNT1" "DEC1"   "GIF"    "LOR"    "MUM1"   "NOV"    "QARS" "SARS"   "SEPT2" 


compare_both = compare2 %in% compare1
table(compare_both) #549 FALSE, 9 TRUE so 9 of them overlap

compare_both = subset(Wilkerson_not_matching_genes, compare1 %in% compare2)
compare_both #[1] "ATAD4"     "C14orf159" "C19orf22"  "C9orf114"  "CCBP2"    
# "CCDC99"    "CPSF3L"    "CTGF"      "CXorf15"   "FAM59A"   
# "HIST1H2AB" "HNRPDL"    "LIN28"     "MAK10"     "MGC13053" 
# "MUPCDH"    "NAG18"     "NEURL"     "ProSAPiP1" "SPINLW1"  
# "TUBB4"    


#unique from symbols
unique_compare1 = subset(compare1, !(compare1 %in% compare2))
unique_compare1
#combine for a total covered list
pot_target_gene_names = c(unique_compare1, compare2)
length(pot_target_gene_names) #923
#check for duplicates
table(duplicated(pot_target_gene_names)) #923 FALSE

unmatched_wilkerson = subset(Wilkerson_not_matching_genes, !(Wilkerson_not_matching_genes %in% pot_target_gene_names))
length(unmatched_wilkerson) #33


#need to change names for Wilkerson_not_matching genes 

#also need to check gene names for RNAseq 

head(MatchedCellsTable)
dim(MatchedCellsTable)
write.csv(MatchedCellsTable, file = "~/Documents/CCLE_LUSC/Exports/MatchedCellsTable_CCLERNASeq_reads.csv")
```


#Change Wilkerson alias to symbol names
```{r}
head(wilkerson_centroid)
colnames(wilkerson_centroid)
dim(wilkerson_centroid)
#issue that column 1 has the gene names and don't know how to target it - row names is the gene alias - read in data again and not put row.names = TRUE
wilkerson_centroid_mappedgenes = wilkerson_centroid

head(wilkerson_centroid_mappedgenes)
#row.names(wilkerson_centroid_mappedgenes)[1] = "Ashley" ---this changed the first row to name Ashley
  

#first check to see if wilkerson gene centroid names match CCLE RNAseq, 
#if not then check alias table and see if alias match and change to gene symbol name
head(Mapping_table_symbol_alias_long) #alias table
head(CCLERNAseq_colnames) #gene names from the CCLE RNAseq 

for (i in 1:nrow(wilkerson_centroid_mappedgenes)) {
  print(i)
  current_gene = row.names(wilkerson_centroid_mappedgenes)[i]
  # need if statement here; does the current_gene match CCLE?
  if (!current_gene %in% CCLERNAseq_colnames) {
    # does not match, need to get aliases for this gene symbol
    if (current_gene %in% Mapping_table_symbol_alias_long$Alias) {
      # gets the alias to gene symbol mapping
      current_gene_aliases = subset(Mapping_table_symbol_alias_long, Alias == current_gene)
      for (j in 1:nrow(current_gene_aliases)) {
        # check if the mapping/new gene symbol is already in Wilkerson
        if (!current_gene_aliases$symbol[j] %in% row.names(CCLERNAseq_colnames)) {
        # assigns the mapping back into the wilkerson_centroid_mappedgenes data frame
        row.names(wilkerson_centroid_mappedgenes)[i] = current_gene_aliases$symbol
      }
      }
    }
  }
}
#stopped at 702 because duplicate row.names not allowed

head(wilkerson_centroid)
dim(wilkerson_centroid) #9361
wilkerson_centroid_mappedgenes = wilkerson_centroid[-c(702,856, 920, 1121, 2875, 3070, 3268, 4546, 5134, 5221, 5452, 5784, 6616, 7300),]
#duplicate at 856, 920, 1121, 2875, 3070, 3268, 4546, 5134, 5221, 5452, 5784, 6616, 7300
dim(wilkerson_centroid_mappedgenes) #9347


for (i in (1:nrow(wilkerson_centroid_mappedgenes))) {
  print(i)
  current_gene = row.names(wilkerson_centroid_mappedgenes)[i]
  # need if statement here; does the current_gene match CCLE?
  if (!current_gene %in% CCLERNAseq_colnames) {
    # does not match, need to get aliases for this gene symbol
    if (current_gene %in% Mapping_table_symbol_alias_long$Alias) {
      # gets the alias to gene symbol mapping
      current_gene_aliases = subset(Mapping_table_symbol_alias_long, Alias == current_gene)
      for (j in 1:nrow(current_gene_aliases)) {
        # check if the mapping/new gene symbol is already in Wilkerson
        if (!current_gene_aliases$symbol[j] %in% row.names(CCLERNAseq_colnames)) {
        # assigns the mapping back into the wilkerson_centroid_mappedgenes data frame
        row.names(wilkerson_centroid_mappedgenes)[i] = current_gene_aliases$symbol
      }
      }
    }
  }
}

#duplicate at 856, 920, 1121, 2875, 3070, 3268, 4546, 5134, 5221, 5452, 5784, 6616, 7300

dim(wilkerson_centroid_mappedgenes)
write.csv(wilkerson_centroid_mappedgenes, file = "~/Documents/CCLE_LUSC/Exports/wilkerson_centroid_mappedgenes_edited9347.csv")

```


#HAVENT FINISHED- run Wilkerson
```{r}
#Goal compare gene names from Wilkerson centroid to gene names from RNASeq
compare = row.names(wilkerson_centroid_mappedgenes) %in% CCLERNAseq_colnames
table(compare) #False 448 (before 956) and True 8899 (before 8405) sooo 956 of the Wilkerson genes don't match what is in the CCLE RNASeq
compare_genes = Wilkerson_genes[compare]

compare2 = CCLERNAseq_colnames %in%  row.names(wilkerson_centroid_mappedgenes)
table(compare2) #false 45458 and true 8899





names(wilkerson_centroid)

gene_df_for_wilkerson = wilkerson_centroid_mappedgenes
head(gene_df_for_wilkerson)
#row.names(gene_df_for_wilkerson) = row.names(gene_df_for_wilkerson$Symbol
genes_for_wilkerson = intersect(row.names(wilkerson_centroid_mappedgenes), row.names(counts))
length(genes_for_wilkerson) #8899

gene_wilkerson_corrs = cor(counts[genes_for_wilkerson,], wilkerson_centroid[genes_for_wilkerson, ])
head(gene_wilkerson_corrs)

genes_for_wilkerson$wilkerson_classifier = apply(gene_wilkerson_corrs, 1, function(x) { names(x)[which.max(x)]}) #says coercing into a list

head(genes_for_wilkerson)
`#rm(gene_df_for_wilkerson)

# 

wilkerson_centroid = read.csv("data/wilkerson.2012.LAD.predictor.centroids.txt", row.names = 1)
names(wilkerson_centroid) = c("Terminal Respiratory Unit", "Proximal-Proliferative", "Proximal-Inflammatory")
gene_df_for_wilkerson = gene
row.names(gene_df_for_wilkerson) = gene_df_for_wilkerson$Symbol
genes_for_wilkerson = intersect(row.names(MatchedCellsTable), row.names(wilkerson_centroid))
gene_wilkerson_corrs = cor(MatchedCellsTable[genes_for_wilkerson, prot_meta_lung$BARCODE], wilkerson_centroid[genes_for_wilkerson, ])
prot_meta_lung$wilkerson_classifier = apply(gene_wilkerson_corrs, 1, function(x) { names(x)[which.max(x)]})
rm(gene_df_for_wilkerson)



```


#HAVENT FINISHED expand and organize sample_meta
```{r}

#subtype using cellosauric_NCIt ... for Lung squamous cell carcinoma
sample_meta_LUSC = subset(sample_meta, sample_meta$Cellosaurus_NCIt_disease == "Lung squamous cell carcinoma")
dim(sample_meta_LUSC) #35 x 33

colnames(sample_meta_LUSC)[colnames(sample_meta_LUSC) == "Predicted"] = "WPredicted"

#Add Stewart subtypes




#add columns for classical or not (WClassical or SClassical ) answer with Y/N
WPredicted_df = data.frame(sample_meta_LUSC[, c("DepMap_ID","WPredicted")])
WPredicted_df = na.omit(WPredicted_df)
WClassical = WPredicted_df$WPredicted == "Classical"
WClassical
WPredicted_df$WClassical = WClassical
WPredicted_df = WPredicted_df[,-2]
WPredicted_df
sample_meta_LUSC = left_join(sample_meta_LUSC, WPredicted_df)
colnames(sample_meta_LUSC)

dim(sample_meta_LUSC) #35 x 34
sample_meta_LUSC

```


#CCLE_RNAseq_reads data
```{r}
dim(MatchedCellsTable) #1406 54357
head(MatchedCellsTable)

hist(apply(MatchedCellsTable,1, max))
#counts should be in the thousands to millions - SO GOOD




head(MatchedCellsTable) #row names are cells and colnames are genes
dim(MatchedCellsTable) #28 x 54357
counts = t(MatchedCellsTable) #transform so genes are row names and cells are column
dim(counts) #54357 28

counts_na = counts
counts_na[counts_na == 0] = NA
boxplot(log2(counts_na)) #median looks about the same across- likely normalized
```

#DEGs -- generate classical vs not classical groups

```{r}
ClassicalLUSCcells #7 as stripped cell line names vector
NotClassicalLUSCcells #19


row.names(MatchedCellsTable)
table(row.names(MatchedCellsTable) %in% ClassicalLUSCcells) #21 false and 7  are true
table(row.names(MatchedCellsTable) %in% NotClassicalLUSCcells) #14 false and 14 true
#down to 11 cells why?
table(row.names(MatchedCellsTable) %in% cellIDS$DepMap_ID) #13 are true

LUSC_classical = subset(MatchedCellsTable, row.names(MatchedCellsTable) %in% ClassicalLUSCcells_ID)
head(LUSC_classical)
dim(LUSC_classical) # 5 54357

LUSC_Notclassical = subset(MatchedCellsTable, !row.names(MatchedCellsTable) %in% ClassicalLUSCcells_ID)
head(LUSC_Notclassical)
dim(LUSC_Notclassical) #8 x 54357


#next time keep all and sort out metastasis vs 

```

#put into DGEList
```{r}

head(counts)
dim(counts) #54357    24

DGELUSC_CCLE = DGEList(counts, lib.size = colSums(counts), 
        norm.factors = rep(1,ncol(counts)), samples = NULL, group =
          colnames(counts), genes = NULL, remove.zeros = FALSE)
head(DGELUSC_CCLE)


colnames(LUSC_sample_meta)
dim(DGELUSC_CCLE$samples) #24 x 3

DGE_samples_group= DGELUSC_CCLE$samples$group
DGE_samples_group = data.frame(DGE_samples_group)
colnames(DGE_samples_group)[1] = "DepMap_ID" 
DGE_samples_group = left_join(DGE_samples_group, LUSC_sample_meta)

DGE_samples_group


#add group descriptor to the list
DGELUSC_CCLE$samples$Subtype = DGE_samples_group$Predicted
DGELUSC_CCLE



```


```{r, echo = FALSE}
# TMM normalization

DGELUSC_CCLE = calcNormFactors(DGELUSC_CCLE, method = "TMM")
DGELUSC_CCLE


#remove lowly expresssed genes
#table(rowSums(DGELUSC_CCLE$counts==0)==9)
# FALSE 52110 and  TRUE 1860 

keep.exprs <- filterByExpr(DGELUSC_CCLE, group=DGELUSC_CCLE$samples$Subtype)
table(keep.exprs) #false 30964 and true 23393

DGELUSC_CCLE_filtered <- DGELUSC_CCLE[keep.exprs,, keep.lib.sizes=FALSE]
dim(DGELUSC_CCLE_filtered) #21297 x 13
head(DGELUSC_CCLE_filtered$counts)
```



```{r, echo = FALSE}
# ovyrs group design
DGELUSC_CCLE
DGELUSC_CCLE$design = model.matrix(~DGELUSC_CCLE$samples$Subtype)
DGELUSC_CCLE


```

#Voom differential expression using Limma

```{r, echo = FALSE, fig.width = 6, fig.height = 4}
DGELUSC_CCLE_voomed = voom(DGELUSC_CCLE, design = DGELUSC_CCLE$design, plot = TRUE)

library(ggbiplot)
rnaseq_pca = prcomp(t(DGELUSC_CCLE_voomed$E), center = TRUE, scale = TRUE)
head(rnaseq_pca)

labels = colnames(rnaseq_pca$sd)
rnaseq_pca$x


ggbiplot(rnaseq_pca, obs.scale = 1, var.scale = 1, labels = rownames(rnaseq_pca$x),
         ellipse = FALSE, circle = TRUE, var.axes = FALSE, groups = DGELUSC_CCLE$samples$Subtype) +
    geom_point(aes(color = DGELUSC_CCLE$samples$Subtype), size = 2, alpha = 0.5)
    scale_color_discrete(name = '') 
    #theme(legend.direction = 'horizontal', legend.position = 'top')
    
#also grouping by adenosquamous, metastasis vs primary, etc - change to cell line name upstream instead of DepMap_ID


```
#Voom fit generate Bayes table
```{r, echo = FALSE}
DGELUSC_CCLE_voomed_vfit = lmFit(DGELUSC_CCLE_voomed, DGELUSC_CCLE_voomed$design)
DGELUSC_CCLE_voomed_efit = eBayes(DGELUSC_CCLE_voomed_vfit)
# ovyrs_group_ebayes_results_table = topTable(ovyrs_group_dge_efit, number = nrow(ovyrs_group_dge_voomed$E)) 
DGELUSC_CCLE_ebayes_results_table = topTable(DGELUSC_CCLE_voomed_efit, number = nrow(DGELUSC_CCLE_voomed$E), coef = 2)


head(DGELUSC_CCLE_ebayes_results_table)

#write.csv(DGELUSC_CCLE_ebayes_results_table, "~/Documents/Rstudio/LUSC/graphs/CCLE_RNAseq_voomTable.csv")
#nrow(subset(ovyrs_group_ebayes_results_table, adj.P.Val <= 0.1))


#look at adj.P.Val <0.05 -- goal is dozens to hundreds of genes
# put into enrichR then 
```
```{r}

increasing = subset(DGELUSC_CCLE$counts[row.names(DGELUSC_CCLE_top7_hits),])
write.csv(increasing, "~/Documents/Rstudio/LUSC/graphs/top6increasing.csv")

decreasing = subset(DGELUSC_CCLE$counts[row.names(CCLE_sig_up),])
decreasing
write.csv(decreasing, "~/Documents/Rstudio/LUSC/graphs/top6decreas==ing.csv")

DGELUSC_CCLE[1,]
```

#Top limma results; p-value distribution

```{r, echo = FALSE}
head(DGELUSC_CCLE_ebayes_results_table)
hist = hist(DGELUSC_CCLE_ebayes_results_table$P.Value, breaks = 25, xlab = "P-value", main = "",  ylab = "Frequency")


dim(subset(DGELUSC_CCLE_ebayes_results_table, logFC <= -1  & adj.P.Val <= 0.05)) #66 by 6
row.names(subset(DGELUSC_CCLE_ebayes_results_table, logFC <= -1  & adj.P.Val <= 0.05))
CCLE_sig_down = subset(subset(DGELUSC_CCLE_ebayes_results_table, logFC <= -1  & adj.P.Val <= 0.05))
dim(CCLE_sig_down) #66 x 6
write.csv(CCLE_sig_down, "~/Documents/Rstudio/LUSC/graphs/CCLE_Down_significantgenes.csv")


CCLE_sig_up = subset(subset(DGELUSC_CCLE_ebayes_results_table, logFC >= 1  & adj.P.Val <= 0.05))
dim(CCLE_sig_up) #7 x 6
write.csv(CCLE_sig_up, "~/Documents/Rstudio/LUSC/graphs/CCLE_up_significantgenes.csv")
row.names(subset(DGELUSC_CCLE_ebayes_results_table, logFC >= 1  & adj.P.Val <= 0.05))
paste0(row.names(subset(DGELUSC_CCLE_ebayes_results_table, logFC >= 1  & adj.P.Val <= 0.05)), collapse = ",")

paste0(row.names(subset(DGELUSC_CCLE_ebayes_results_table, logFC <= 0  & adj.P.Val <= 0.05)), collapse = ",")




#volcano plot
plot(DGELUSC_CCLE_ebayes_results_table$logFC, -log10(DGELUSC_CCLE_ebayes_results_table$P.Value))

# write.table(ovyrs_group_ebayes_results_table, "work/ovyrs_group_ebayes_results_table_2020-08-31.txt", sep = "\t", quote = F, row.names = F)

head(DGELUSC_CCLE_ebayes_results_table)
DGELUSC_CCLE_ebayes_results_table2 = DGELUSC_CCLE_ebayes_results_table
DGELUSC_CCLE_ebayes_results_table2 = mutate(DGELUSC_CCLE_ebayes_results_table2, neglog10pvalue = -log10(P.Value))
head(DGELUSC_CCLE_ebayes_results_table2)

library(RColorBrewer)
library(ggrepel)

ggplot(data = DGELUSC_CCLE_ebayes_results_table2, aes (x = logFC, y = neglog10pvalue)) +
  geom_vline(xintercept = c(-2, 2), linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), linetype = 'dashed') + 
  geom_point()
  

```

Plots of top hits if they exist

```{r, echo = FALSE, include = FALSE}
head(DGELUSC_CCLE_ebayes_results_table)
DGELUSC_CCLE_top6_hits = DGELUSC_CCLE_ebayes_results_table[1:6, ]
DGELUSC_CCLE_top6_hits

head(DGELUSC_CCLE_voomed$E)
DGELUSC_CCLE_top6_hits_df = DGELUSC_CCLE_voomed$E[row.names(DGELUSC_CCLE_top6_hits), ]

DGELUSC_CCLE_top7_hits = DGELUSC_CCLE_ebayes_results_table[1:7, ]
DGELUSC_CCLE_top7_hits

dim(DGELUSC_CCLE_ebayes_results_table) 

par(mfrow = c(2,3))
for (i in 1:nrow(DGELUSC_CCLE_top6_hits_df)) {
  boxplot(row.name(DGELUSC_CCLE_top6_hits)[i]
          }

boxplot(list(quartile1 = ovyrs_group_top6_hits_df[ovyrs_group_top6_hits$GencodeID[i], quartile1_patients], quartile2 = ovyrs_group_top6_hits_df[ovyrs_group_top6_hits$GencodeID[i], quartile2_patients], quartile3 = ovyrs_group_top6_hits_df[ovyrs_group_top6_hits$GencodeID[i], quartile3_patients], quartile4 = ovyrs_group_top6_hits_df[ovyrs_group_top6_hits$GencodeID[i], quartile4_patients]), ylab = "Normalized Expression", main = ovyrs_group_top6_hits$GeneName[i], names = c("Q1", "Q2", "Q3", "Q4"))

```


Plot top hits in heatmap (padj < 0.1)

```{r, echo = FALSE, fig.width = 10, fig.height = 6, warning = FALSE, message = FALSE}
# gene heatmap ----

DGELUSC_CCLE_heatmap_genes = subset(DGELUSC_CCLE_ebayes_results_table, adj.P.Val <= 0.1)
DGELUSC_CCLE_heatmap_df = DGELUSC_CCLE_voomed$E[row.names(DGELUSC_CCLE_heatmap_genes), ]

DGELUSC_CCLE_heatmap_df_transform = as.matrix(t(scale(t(DGELUSC_CCLE_heatmap_df), scale = TRUE, center = TRUE)))
head(DGELUSC_CCLE_heatmap_df_transform)
dim(DGELUSC_CCLE_heatmap_df_transform)
#row.names(DGELUSC_CCLE_heatmap_df_transform) = row.names(DGELUSC_CCLE_heatmap_genes)

library(ggbiplot)
# range(clinical_meta_ovyrs$ovyrs)
DGELUSC_CCLE_col = colorRamp2(c(12, 32, 52), c("white", "lightblue", "darkblue"))

?HeatmapAnnotation
#column_anno = HeatmapAnnotation(
    #ovyrs = clinical_meta_ovyrs$ovyrs,
    #col = list("ovyrs" = DGELUSC_CCLE_col)
)

# create heatmap and draw
gene_heatmap = Heatmap(DGELUSC_CCLE_heatmap_df_transform, cluster_columns = TRUE, cluster_rows = TRUE, row_title = "Gene Expression", column_title = "CCLE Cell lines", name = "Z-score", heatmap_legend_param = list(title_position = "topcenter", legend_direction = "vertical"), show_row_names = FALSE, row_names_gp = gpar(fontsize = 8), show_column_names = FALSE)
# draw(gene_heatmap, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legend = TRUE)
gene_heatmap


head(DGELUSC_CCLE_heatmap_df_transform)


```

\newpage
### Gene set enrichment analysis (Hallmarks of Cancer, Reactome, KEGG)

```{r, echo = FALSE, warning = FALSE, message = FALSE}
## gsea
ovyrs_cont_gsea_ranks = -log10(ovyrs_cont_ebayes_results_table$P.Value)
names(ovyrs_cont_gsea_ranks) = ovyrs_cont_ebayes_results_table$GeneName
```

Hallmarks

```{r, echo = FALSE, warning = FALSE, message = FALSE}
ovyrs_cont_hallmarks = as.data.frame(fgsea(hallmarks, ovyrs_cont_gsea_ranks, minSize = 15, maxSize = 500, scoreType = "pos"))
ovyrs_cont_hallmarks_top10 = head(ovyrs_cont_hallmarks[order(ovyrs_cont_hallmarks$padj), ], 10)
plotGseaTable(hallmarks[ovyrs_cont_hallmarks_top10$pathway], ovyrs_cont_gsea_ranks, ovyrs_cont_hallmarks)
```

\newpage
Reactome

```{r, echo = FALSE, warning = FALSE, message = FALSE}
ovyrs_cont_reactome = as.data.frame(fgsea(reactome, ovyrs_cont_gsea_ranks, minSize = 15, maxSize = 500, scoreType = "pos"))
ovyrs_cont_reactome_top10 = head(ovyrs_cont_reactome[order(ovyrs_cont_reactome$padj), ], 10)
plotGseaTable(reactome[ovyrs_cont_reactome_top10$pathway], ovyrs_cont_gsea_ranks, ovyrs_cont_reactome)
```

\newpage
KEGG

```{r, echo = FALSE, warning = FALSE, message = FALSE}
ovyrs_cont_kegg = as.data.frame(fgsea(kegg, ovyrs_cont_gsea_ranks, minSize = 15, maxSize = 500, scoreType = "pos"))
ovyrs_cont_kegg_top10 = head(ovyrs_cont_kegg[order(ovyrs_cont_kegg$padj), ], 10)
plotGseaTable(kegg[ovyrs_cont_kegg_top10$pathway], ovyrs_cont_gsea_ranks, ovyrs_cont_kegg)
```

Saving limma topTable (file name is endpoint_covariate1+covariate2+..._date) to 

`ovyrs_cont_birthDecade+agedx+cohort_2020-10-23.csv` 
 
standard errors to 
 
`ovyrs_cont_ebayes_SE_2020-10-23.csv` 

GSEA Hallmark results to 

`ovyrs_cont_hallmarks_2020-10-23.csv`

and GSEA Reactome results to 

`ovyrGroup_reactome_2020-10-23.csv`

and GSEA KEGG results to 

`ovyrGroup_kegg_2020-10-23.csv`



```{r, echo = FALSE}
ovyrs_cont_hallmarks$leadingEdge = unlist(lapply(ovyrs_cont_hallmarks$leadingEdge, function(x) {
    unlisted_strip = str_replace_all(x, fixed("\""), "")
    unlisted_strip = str_replace_all(unlisted_strip, "c\\(|\\)|\\\n", "")
    paste(unlisted_strip, collapse = ",")
}))

ovyrs_cont_reactome$leadingEdge = unlist(lapply(ovyrs_cont_reactome$leadingEdge, function(x) {
    unlisted_strip = str_replace_all(x, fixed("\""), "")
    unlisted_strip = str_replace_all(unlisted_strip, "c\\(|\\)|\\\n", "")
    paste(unlisted_strip, collapse = ",")
}))

ovyrs_cont_kegg$leadingEdge = unlist(lapply(ovyrs_cont_kegg$leadingEdge, function(x) {
    unlisted_strip = str_replace_all(x, fixed("\""), "")
    unlisted_strip = str_replace_all(unlisted_strip, "c\\(|\\)|\\\n", "")
    paste(unlisted_strip, collapse = ",")
}))

```

```{r, echo = FALSE, warning = FALSE, message = FALSE}
write.table(ovyrs_cont_ebayes_results_table, "work/ovyrs_cont_birthDecade+agedx+cohort_2020-10-23.csv", row.names = FALSE, sep = ",", quote = T)
ovyrs_cont_se = sqrt(ovyrs_cont_dge_efit$s2.post) * ovyrs_cont_dge_efit$stdev.unscaled
write.table(ovyrs_cont_se, "work/ovyrs_cont_ebayes_SE_2020-10-23.csv", row.names = TRUE, sep = ",", quote = T)

write.table(ovyrs_cont_hallmarks, "work/ovyrs_cont_hallmarks_2020-10-23.csv", row.names = F, sep = ",", quote = T)
write.table(ovyrs_cont_reactome, "work/ovyrs_cont_reactome_2020-10-23.csv", row.names = F, sep = ",", quote = T)
write.table(ovyrs_cont_kegg, "work/ovyrs_cont_kegg_2020-10-23.csv", row.names = F, sep = ",", quote = T)
```







#Old DESeq code
```{r}
CN <- round(CCLEtranscript, digits=0)

df <- CN


### Check the distribution
par(mfrow = c(2,2), mar = c(4,4,1,1))
print( apply(df, 2, function(x) quantile(as.numeric(x))) )
log1 <- apply(df, 2, function(x) {hist(log2(x), breaks = 100)})


idExp <- apply(df, 1, function(x) any(x > 16))
dfExp <- df[idExp, ]
print(dim(dfExp))

library('DESeq2')

table2 <- data.frame(name = colnames(dfExp),
                     condition = CCLE_LUSC_meta$wilkerson_classifier)
dds <- DESeqDataSetFromMatrix(dfExp, 
                              colData=table2, design= ~ condition)
dds <- DESeq(dds)

norCounts <- counts(dds, normalized=TRUE)

res <- results(dds)

head(res)
print(res)
writexl(res, "")

## extract the genes with adjP < 0.0001
resSig = res[ which(res$padj < 0.01), ]
dim(resSig)

sigNorData <- norCounts[which(res$padj < 0.01),]

library(RColorBrewer)
library(gplots)
library(ggplot2)
hmcol =  colorRampPalette(brewer.pal(9, "GnBu"))(100)


heatMapDF_nor <- t( apply(sigNorData, 1, function(x){(x-mean(x))/sd(x)}) )

colnames(heatMapDF_nor) <- c('Young_1','Young_2','Young_3','Young_4',
                             'Aged_1','Aged_2','Aged_3','Aged_4',
                             'AgedNPC86_1','AgedNPC86_2','AgedNPC86_3','AgedNPC86_4')

p <- heatmap.2(heatMapDF_nor, col = hmcol, trace="none", margin=c(10, 10),labRow = F)

heatmap.2(heatMapDF_nor, Rowv=T, Colv=FALSE, dendrogram ="row", #scale ="row", col = col,
          density.info="none", trace="none", margins = c(10,10))
pdf('heatmap.pdf')
p
# to find how many with fold change greater than 1 or less than -1

length(which(resSig$log2FoldChange > 1))
length(which(resSig$log2FoldChange < -1))


res_plot      <- data.frame( res )
res_plot$col  <- 'gray40'
res_plot$col[res_plot$log2FoldChange > 1 & res_plot$padj < 0.01] <- 'red'
res_plot$col[res_plot$log2FoldChange < -1 & res_plot$padj < 0.01] <- 'cornflowerblue'

plot(res_plot$log2FoldChange,
     -log10(res_plot$padj),
     col = res_plot$col, pch = 19, xlab = 'log2(fold change)',
     ylab = '-log10(p-adj)', 
     las = 1 # the direction of ylab will be change
)


```