---
title: "LUSC_MoffittPtSamples"
author: "AL"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
#Load libraries and setwd
library(here)
library(readxl)
library(stringr)
library(ConsensusClusterPlus)
library(ComplexHeatmap)
library(circlize)
library(enrichR)
library(immunedeconv)
library(ggplot2)
library(iClusterPlus)
library(dplyr)
library(tidyr)
library(reshape2)
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 60), tidy = TRUE)
set.seed(2022)

getwd()
#setwd("smb://hlm/data/project/LungCancerMetabolismProgram")

```

## Find levels of p63


```{r pressure, echo=FALSE}
transcript = read.delim("~/Documents/Rstudio/LUSC/RSEM.isoform.expected_count.txt", stringsAsFactors = FALSE)
# do i match tp63?

colnames(transcript)
#see all 108ish samples

#??str_detect
#like grep


table(str_detect(transcript$GeneName, "TP63"))
# how many TP63 in GeneName column? answer is 14

# just get the tp63 from the transcript data frame
transcript_tp63 = transcript[str_detect(transcript$GeneName, "TP63"), ]

#see all rows and first 5 columns
transcript_tp63[,1:5]


patient_meta = read.delim("~/Documents/Rstudio/LUSC/tmt_meta_up_to_proteomic_heatmaps_2019-10-10.txt", header = TRUE, stringsAsFactors = FALSE)


# look at just patient data
#transcript_tp63[, patient_meta$GenomicsSample]

# uh oh, the above gave an error; where is the problem?
head(patient_meta$GenomicsSample)
str(patient_meta$GenomicsSample)
#108 samples

names(transcript_tp63) %in% patient_meta$GenomicsSample
# here are the ones that do not match

Matched = names(transcript_tp63) %in% patient_meta$GenomicsSample
table(Matched)
#13 don't match - why total 119 counts?

names(transcript_tp63)[!names(transcript_tp63) %in% patient_meta$GenomicsSample]
# actually look at the flip since we care about what is in the patient meta table
#What RNAseq samples are not in meta samples?


length(intersect(patient_meta$GenomicsSample,names(transcript_tp63)))
#106 match


patient_meta$GenomicsSample[!patient_meta$GenomicsSample %in% names(transcript_tp63)]
# turns out there mismatches because of the "redo" text
#What meta samples are not in the RNAseq samples - 2 redos


patient_meta$GenomicsSampleSplicing = str_replace(patient_meta$GenomicsSample, "Redo", "")
# did it work? like ggsub to replace part of a name with something else in meta data into new columm - don't want to overwrite original name
length(patient_meta$GenomicsSampleSplicing[patient_meta$GenomicsSampleSplicing %in% names(transcript_tp63)])
# hurray!
# look at the table
transcript_tp63[, patient_meta$GenomicsSampleSplicing]
# make a boxplot the BAD way
#boxplot(as.numeric(transcript_tp63[3, patient_meta$GenomicsSampleSplicing]))
transcript_tp63

write.csv(transcript_tp63, file="~/Documents/Rstudio/LUSC/LUSC Stewart human pt tumors rough boxplots/Exported_TP63levels.csv")
?write.csv


row.names(transcript_tp63)
# the row names are not very descriptive; lets fix this

#TP63names <- c("TAp63alpha","TAp63delta","dNp63alpha","TAp63beta","dNp63delta","dNp63beta","TAp63gamma", "14261","deltaNp63gamma", "p63-delta", "deltaNp73L", "14265", "14266","14267")

row.names(transcript_tp63) = transcript_tp63$GeneName
row.names(transcript_tp63)


transcript_tp63["TP63-203", patient_meta$GenomicsSampleSplicing]



redox_patients = subset(patient_meta, subtype == "Redox")$GenomicsSampleSplicing
inflamed_patients = subset(patient_meta, subtype == "Inflamed")$GenomicsSampleSplicing
colnames(patient_meta)


tp63_203_redox = as.numeric(transcript_tp63["TP63-203", redox_patients])
tp63_203_inflamed = as.numeric(transcript_tp63["TP63-203", inflamed_patients])

#deltaNp63alpha is 203
tp63_203_redox = log2(as.numeric(transcript_tp63["TP63-203", redox_patients]))
tp63_203_inflamed = log2(as.numeric(transcript_tp63["TP63-203", inflamed_patients]))

# these are not so helpful separately; do them together with a list
boxplot(list("Inflamed" = tp63_203_inflamed, "Redox" = tp63_203_redox),
        main = "deltaNp63alpha",
        xlab = "Subtype", ylab = "fpkm?")



length(redox_patients)
length(inflamed_patients)

getwd()
#"/Users/4476306/Documents/Rstudio/LUSC/scripts/Scripts"

setwd("/Users/4476306/Documents/Rstudio/LUSC/")
Pt_Summed_TP63_isoforms = read.csv("/Users/4476306/Documents/Rstudio/LUSC/Pt_RNAseq_TP63_summed.csv", stringsAsFactors = FALSE, check.names=FALSE)
Pt_Summed_TP63_isoforms


#separate redox tumors
df_redox_summed = as.data.frame(redox_patients)
df_redox_summed

df_redox_summed = left_join(df_redox_summed, Pt_Summed_TP63_isoforms, by = c("redox_patients" =  "Pt_Info"))
write.csv(df_redox_summed, file="/Users/4476306/Documents/Rstudio/LUSC/redox_summed.csv")


#separate inflamed tumors
df_inflamed_summed = as.data.frame(inflamed_patients)
df_inflamed_summed

df_inflamed_summed = left_join(df_inflamed_summed, Pt_Summed_TP63_isoforms, by = c("inflamed_patients" =  "Pt_Info"))
write.csv(df_inflamed_summed, file="/Users/4476306/Documents/Rstudio/LUSC/inflamed_summed.csv")
```

#make box plots

```{r pressure, echo=FALSE}
?boxplot
# for ashley to do:
# for (i in 1:nrow(transcript_tp63)) {
# 	tp63_redox = log2(as.numeric(transcript_tp63[i, redox_patients]))
# tp63_inflamed = log2(as.numeric(transcript_tp63[i, inflamed_patients]))
# t.test()
# }


tp63_201_redox = as.numeric(transcript_tp63["TP63-201", redox_patients])
tp63_201_inflamed = as.numeric(transcript_tp63["TP63-201", inflamed_patients])

#TAp63alpha is 201
tp63_201_redox = log2(as.numeric(transcript_tp63["TP63-201", redox_patients]))
tp63_201_inflamed = log2(as.numeric(transcript_tp63["TP63-201", inflamed_patients]))

# these are not so helpful separately; do them together with a list
boxplot(list("Inflamed" = tp63_201_inflamed, "Redox" = tp63_201_redox),
        main = "TAp63alpha",
        xlab = "Subtype", ylab = "log2")







tp63_204_redox = as.numeric(transcript_tp63["TP63-204", redox_patients])
tp63_204_inflamed = as.numeric(transcript_tp63["TP63-204", inflamed_patients])

#TAp63beta is 204
tp63_204_redox = log2(as.numeric(transcript_tp63["TP63-204", redox_patients]))
tp63_204_inflamed = log2(as.numeric(transcript_tp63["TP63-204", inflamed_patients]))

# these are not so helpful separately; do them together with a list
boxplot(list("Inflamed" = tp63_204_inflamed, "Redox" = tp63_204_redox),
        main = "TAp63beta",
        xlab = "Subtype", ylab = "log2")






tp63_206_redox = as.numeric(transcript_tp63["TP63-206", redox_patients])
tp63_206_inflamed = as.numeric(transcript_tp63["TP63-206", inflamed_patients])

#deltaNp63beta is 206
tp63_206_redox = log2(as.numeric(transcript_tp63["TP63-206", redox_patients]))
tp63_206_inflamed = log2(as.numeric(transcript_tp63["TP63-206", inflamed_patients]))

# these are not so helpful separately; do them together with a list
dNp63_boxplot <- boxplot(list("Inflamed" = tp63_206_inflamed, "Redox" = tp63_206_redox),
        main = "deltaNp63beta", xlab = "Subtype", ylab = "fpkm") 

dNp63_boxplot + stat_pvalue_manual(dNp63beta, label = "p") + scale_y_continuous(expand = expansion(mult = c(0.05, 0.1)))

dNp63beta <- t.test(tp63_206_inflamed,tp63_206_redox)







tp63_207_redox = as.numeric(transcript_tp63["TP63-207", redox_patients])
tp63_207_inflamed = as.numeric(transcript_tp63["TP63-207", inflamed_patients])

#TAp63gamma is 207
tp63_207_redox = log2(as.numeric(transcript_tp63["TP63-207", redox_patients]))
tp63_207_inflamed = log2(as.numeric(transcript_tp63["TP63-207", inflamed_patients]))

# these are not so helpful separately; do them together with a list
boxplot(list("Inflamed" = tp63_207_inflamed, "Redox" = tp63_207_redox),
        main = "TAp63gamma",
        xlab = "Subtype", ylab = "log2")






tp63_209_redox = as.numeric(transcript_tp63["TP63-209", redox_patients])
tp63_209_inflamed = as.numeric(transcript_tp63["TP63-209", inflamed_patients])

#deltaNp63beta is 206
tp63_209_redox = log2(as.numeric(transcript_tp63["TP63-209", redox_patients]))
tp63_209_inflamed = log2(as.numeric(transcript_tp63["TP63-209", inflamed_patients]))

# these are not so helpful separately; do them together with a list
boxplot(list("Inflamed" = tp63_209_inflamed, "Redox" = tp63_209_redox),
        main = "deltaNp63gamma",
        xlab = "Subtype", ylab = "log2")








#combine all as fpkm
boxplot(list("Inflamed_TAp63alpha" = tp63_201_inflamed, "Redox_TAp63alpha" = tp63_201_redox,
  "Inflamed_deltaNp63alpha" = tp63_206_inflamed, "Redox_deltaNp63alpha" = tp63_206_redox,
        "Inflamed_TAp63beta" = tp63_204_inflamed, "Redox_TAp63beta" = tp63_204_redox,  
  "Inflamed_deltaNp63beta" = tp63_206_inflamed, "Redox_deltaNp63beta" = tp63_206_redox,     
       "Inflamed_TAp63gamma" = tp63_207_inflamed, "Redox_TAp63gamma" = tp63_207_redox,
  "Inflamed_deltaNp63gamma" = tp63_209_inflamed, "Redox_deltaNp63gamma" = tp63_209_redox), srt=45,
       xlab = "Counts (fpkm?)",  horizontal = TRUE )

  
?boxplot


allTP63_asCounts <- list(tp63_201_inflamed, tp63_201_redox,
                        tp63_203_inflamed,tp63_203_redox,
                        tp63_204_inflamed,tp63_204_redox,  
                         tp63_206_inflamed,tp63_206_redox,     
                        tp63_207_inflamed,tp63_207_redox,
                        tp63_209_inflamed, tp63_209_redox) 

boxplot(allTP63_asCounts, horizontal = TRUE, xlab = "counts", main = "Inflamed v Redox in SqCC Tumors by RNASeq", names = axis_labels, las=1, par(mar=c(4,13,2,2)))

axis_labels <- c("Inflamed_TAp63alpha", "Redox_TAp63alpha", "Inflamed_deltaNp63alpha", "Redox_deltaNp63alpha", 
                 "Inflamed_TAp63beta", "Redox_TAp63beta", "Inflamed_deltaNp63beta", "Redox_deltaNp63beta",
                 "Inflamed_TAp63gamma", "Redox_TAp63gamma",  "Inflamed_deltaNp63gamma", "Redox_deltaNp63gamma")





alldeltaNp63_asLogs <- list(tp63_203_inflamed, tp63_203_redox,
                         tp63_206_inflamed,tp63_206_redox) 
delta_axis_labels <- c("Inflamed_deltaNp63alpha", "Redox_deltaNp63alpha", 
                       "Inflamed_deltaNp63beta", "Redox_deltaNp63beta")

boxplot(alldeltaNp63_asLogs, ylab = "counts", main = "Inflamed v Redox in SqCC Tumors by RNASeq", names = delta_axis_labels, las=2, par(mar=c(12,5,2,2))) +
    t.test()

?t.test
