---
title: "Subset_CCLERNASeq"
author: "AL"
date: "2023-08-29"
output: html_document
---

#Load Libraries
```{r setup, include=FALSE}
#Load libraries and setwd
library(here)
library(readxl)
library(stringr)
library(ConsensusClusterPlus)
library(ComplexHeatmap)
library(circlize)
library(enrichR)
library(immunedeconv)
library(ggplot2)
library(iClusterPlus)
library(dplyr)
library(tidyr)
library(reshape2)
library(limma)
library(edgeR)
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 60), tidy = TRUE)
set.seed(2022)

getwd()
```

#Load subset RNASeq data
```{r pressure, echo=FALSE}

CCLE_RNAseq_reads_LUSC = read.csv(file="~/Project_LUSC/RScripts_LUSC/Exports/CCLE_RNAseq_reads_LUSC.csv", header = TRUE, row.names = 1)

CCLEcelllines_classifiedbyWilkerson = read.csv("~/Project_LUSC/RScripts_LUSC/Resources/LUSCclassifers_WilkersonTable2.csv",header=TRUE, stringsAsFactors = FALSE)

sample_meta_LUSCsubtyped = read.csv(file = "~/Project_LUSC/RScripts_LUSC/Exports/sample_meta_LUSCsubtyped.csv", header = TRUE)
#Wilkerson predicted subtypes


```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.





#rename wilkerson subtypes from our predicted and wilkerson paper
```{r}
#Spredicted is our predictions with SClassical being classical or not
colnames(sample_meta_LUSCsubtyped)[colnames(sample_meta_LUSCsubtyped) == "wilkerson_classifier"] = "SPredicted"

SPredicted_df = data.frame(sample_meta_LUSCsubtyped[, c("DepMap_ID","SPredicted")])
SClassical = SPredicted_df$SPredicted == "classical"
SPredicted_df$SClassical = SClassical
SPredicted_df = SPredicted_df[,-2]
sample_meta_LUSCsubtyped = left_join(sample_meta_LUSCsubtyped, SPredicted_df)


#add columns for classical or not by wilkerson paper (WPredicted and WClassical)
WPredicted_df = data.frame(CCLEcelllines_classifiedbyWilkerson[, c("Cell.line","Predicted")])
colnames(WPredicted_df)[colnames(WPredicted_df) == "Cell.line"] = "cell_line_name"
colnames(WPredicted_df)[colnames(WPredicted_df) == "Predicted"] = "WPredicted"
WClassical = WPredicted_df$WPredicted == "Classical"
WPredicted_df$WClassical = WClassical
sample_meta_LUSCsubtyped = left_join(sample_meta_LUSCsubtyped, WPredicted_df)


#only 4 match up by SPredicted and WPredicted -- how to code this? 7 ludlu-1, 16 hcc-2814, 18 lc-1f, 23 hcc15
```


#CCLE_RNAseq_reads data
```{r}
hist(apply(CCLE_RNAseq_reads_LUSC,1, max))
#counts should be in the thousands to millions - SO GOOD


counts = t(CCLE_RNAseq_reads_LUSC) #transform so genes are row names and cells are column
dim(counts) #54357 21

counts_na = counts
counts_na[counts_na == 0] = NA
boxplot(log2(counts_na)) #median looks about the same across- likely normalized
```


#put into DGEList
```{r}

DGELUSC_CCLE = DGEList(counts, lib.size = colSums(counts), 
        norm.factors = rep(1,ncol(counts)), samples = NULL, group =
          colnames(counts), genes = NULL, remove.zeros = FALSE)

DGE_samples_group= DGELUSC_CCLE$samples$group
DGE_samples_group = data.frame(DGE_samples_group)

colnames(DGE_samples_group)[1] = "stripped_cell_line_name" 
DGE_samples_group = left_join(DGE_samples_group, sample_meta_LUSCsubtyped)


#add group descriptor to the list
DGELUSC_CCLE$samples$WPredicted = DGE_samples_group$WPredicted
DGELUSC_CCLE$samples$WClassical = DGE_samples_group$WClassical
DGELUSC_CCLE$samples$SPredicted = DGE_samples_group$SPredicted
DGELUSC_CCLE$samples$SClassical = DGE_samples_group$SClassical
DGELUSC_CCLE$samples$PrimaryMetastatic = DGE_samples_group$primary_or_metastasis
DGELUSC_CCLE$samples$sex = DGE_samples_group$sex
DGELUSC_CCLE$samples$p53 = DGE_samples_group$p53



```


```{r, echo = FALSE}
# TMM normalization

DGELUSC_CCLE = calcNormFactors(DGELUSC_CCLE, method = "TMM")
DGELUSC_CCLE


#remove lowly expresssed genes
table(rowSums(DGELUSC_CCLE$counts==0)==9)
# FALSE 53686 and  TRUE 671 

keep.exprs <- filterByExpr(DGELUSC_CCLE, group=DGELUSC_CCLE$samples$WClassical)
table(keep.exprs) #false 32427  and true 21930 

DGELUSC_CCLE_filtered <- DGELUSC_CCLE[keep.exprs,, keep.lib.sizes=FALSE]
 #21930     x 28

```



```{r, echo = FALSE}
#group design
DGELUSC_CCLE
DGELUSC_CCLE$samples
DGELUSC_CCLE$samples$SClassical


DGELUSC_CCLE$design = model.matrix(~DGELUSC_CCLE$samples$SClassical)
DGELUSC_CCLE


```

#Voom differential expression using Limma

```{r, echo = FALSE, fig.width = 6, fig.height = 4}
DGELUSC_CCLE_voomed = voom(DGELUSC_CCLE, design = DGELUSC_CCLE$design, plot = TRUE)

library(ggbiplot)
rnaseq_pca = prcomp(t(DGELUSC_CCLE_voomed$E), center = TRUE, scale = TRUE)
head(rnaseq_pca)

labels = colnames(rnaseq_pca$sd)
rnaseq_pca$x


ggbiplot(rnaseq_pca, obs.scale = 1, var.scale = 1, labels = rownames(rnaseq_pca$x),
         ellipse = FALSE, circle = TRUE, var.axes = FALSE, groups = DGELUSC_CCLE$samples$SPredicted) +
    geom_point(aes(color = DGELUSC_CCLE$samples$SPredicted), size = 2, alpha = 0.5)
    scale_color_discrete(name = '') 
    #theme(legend.direction = 'horizontal', legend.position = 'top')
    
#also grouping by adenosquamous, metastasis vs primary, etc - change to cell line name upstream instead of DepMap_ID


```
#Voom fit generate Bayes table
```{r, echo = FALSE}
#This analysis is by SClassical

DGELUSC_CCLE_voomed_vfit = lmFit(DGELUSC_CCLE_voomed, DGELUSC_CCLE_voomed$design)
DGELUSC_CCLE_voomed_efit = eBayes(DGELUSC_CCLE_voomed_vfit)
DGELUSC_CCLE_ebayes_results_table = topTable(DGELUSC_CCLE_voomed_efit, number = nrow(DGELUSC_CCLE_voomed$E), coef = 2)


write.csv(DGELUSC_CCLE_ebayes_results_table, "~/Project_LUSC/RScripts_LUSC/Exports/DGELUSC_CCLE_ebayes_results_table.csv")
#nrow(subset(ovyrs_group_ebayes_results_table, adj.P.Val <= 0.1))


#look at adj.P.Val <0.05 -- goal is dozens to hundreds of genes
# put into enrichR then 


hist = hist(DGELUSC_CCLE_ebayes_results_table$P.Value, breaks = 25, xlab = "P-value", main = "",  ylab = "Frequency")
```


# STOPPED HERE --- Top limma results; p-value distribution

```{r, echo = FALSE}



dim(subset(DGELUSC_CCLE_ebayes_results_table, logFC <= -1  & adj.P.Val <= 0.05)) #0 by 6
row.names(subset(DGELUSC_CCLE_ebayes_results_table, logFC <= -1  & adj.P.Val <= 0.05))
CCLE_sig_down = subset(subset(DGELUSC_CCLE_ebayes_results_table, logFC <= -1  & adj.P.Val <= 0.05))
dim(CCLE_sig_down) #25 x 6



CCLE_sig_up = subset(subset(DGELUSC_CCLE_ebayes_results_table, logFC >= 1  & adj.P.Val <= 0.05))
dim(CCLE_sig_up) #38 x 6
row.names(subset(DGELUSC_CCLE_ebayes_results_table, logFC >= 1  & adj.P.Val <= 0.05))


paste0(row.names(subset(DGELUSC_CCLE_ebayes_results_table, logFC >= 1  & adj.P.Val <= 0.05)), collapse = ",")

paste0(row.names(subset(DGELUSC_CCLE_ebayes_results_table, logFC <= 0  & adj.P.Val <= 0.05)), collapse = ",")




#volcano plot
plot(DGELUSC_CCLE_ebayes_results_table$logFC, -log10(DGELUSC_CCLE_ebayes_results_table$P.Value))

colnames(DGELUSC_CCLE_ebayes_results_table)

#generate group and colors for up and down sig genes
DGELUSC_CCLE_ebayes_results_table = DGELUSC_CCLE_ebayes_results_table %>%
  mutate(gene_type = case_when(logFC >= 2 & adj.P.Val <= 0.05 ~ "up",
                               logFC <= 0.5 & adj.P.Val <= 0.05 ~ "down",
                               TRUE ~ "ns"))
count(DGELUSC_CCLE_ebayes_results_table$gene_type)  #25 down 54303 ns and 29 up

DGELUSC_CCLE_ebayes_results_table %>%
  distinct(gene_type) %>%
  pull()  

cols <- c("up" = "#ffad73", "down" = "#26b3ff", "ns" = "grey") 
sizes <- c("up" = 2, "down" = 2, "ns" = 1) 
alphas <- c("up" = 1, "down" = 1, "ns" = 0.5)


library(ggrepel)

DGELUSC_CCLE_ebayes_results_table %>%
  ggplot(aes(x = logFC,
             y = -log10(adj.P.Val),
             fill = gene_type,    
             size = gene_type,
             alpha = gene_type)) + 
  geom_point(shape = 21,   
             colour = "black") + 
  geom_hline(yintercept = -log10(0.05),
             linetype = "dashed") + 
  geom_vline(xintercept = c(log2(0.5), log2(2)),
             linetype = "dashed") +
  scale_fill_manual(values = cols) + 
  scale_size_manual(values = sizes) + 
  scale_alpha_manual(values = alphas) + 
  scale_x_continuous(breaks = c(seq(-10, 10, 2)),       
                     limits = c(-10, 10)) +
  geom_text_repel(data=head(input, 20), aes(label=genes))

?geom_text_repel


#try to label sig genes
AllSigGenes = rbind(CCLE_sig_up, CCLE_sig_down)
dim(AllSigGenes) #63
dim(CCLE_sig_down) #25
dim(CCLE_sig_up) #38

AllSigGenes = as.data.frame(AllSigGenes)

AllSigGenes = mutate(AllSigGenes, genes = row.names(AllSigGenes))


AllSigGenes


ggplot(data = DGELUSC_CCLE_ebayes_results_table,
       aes(x = logFC,
           y = -log10(adj.P.Val))) + 
  geom_point(aes(colour = gene_type), 
             alpha = 0.2, 
             shape = 16,
             size = 1) + 
  geom_point(data = CCLE_sig_up,
             shape = 21,
             size = 2, 
             fill = "firebrick", 
             colour = "black") + 
  geom_point(data = CCLE_sig_down,
             shape = 21,
             size = 2, 
             fill = "steelblue", 
             colour = "black") + 
  geom_hline(yintercept = -log10(0.05),
             linetype = "dashed") + 
  geom_vline(xintercept = c(log2(0.5), log2(2)),
             linetype = "dashed") +
  geom_text_repel (data = AllSigGenes, 
                   mapping = aes(label = genes),
                   force = 2,
                   nudge_y = 1) +
  scale_colour_manual(values = cols) + 
  scale_x_continuous(breaks = c(seq(-10, 10, 2)),     
                     limits = c(-10, 10))  
?geom_text_repel
AllSig_Genes
class(AllSig_Genes)

# write.table(ovyrs_group_ebayes_results_table, "work/ovyrs_group_ebayes_results_table_2020-08-31.txt", sep = "\t", quote = F, row.names = F)

head(DGELUSC_CCLE_ebayes_results_table)
DGELUSC_CCLE_ebayes_results_table2 = DGELUSC_CCLE_ebayes_results_table
DGELUSC_CCLE_ebayes_results_table2 = mutate(DGELUSC_CCLE_ebayes_results_table2, neglog10pvalue = -log10(P.Value))
head(DGELUSC_CCLE_ebayes_results_table2)

library(RColorBrewer)
library(ggrepel)

ggplot(data = DGELUSC_CCLE_ebayes_results_table2, aes (x = logFC, y = neglog10pvalue)) +
  geom_vline(xintercept = c(-2, 2), linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), linetype = 'dashed') + 
  geom_point()
  

```

Plots of top hits if they exist

```{r, echo = FALSE, include = FALSE}
head(DGELUSC_CCLE_ebayes_results_table)
DGELUSC_CCLE_top6_hits = DGELUSC_CCLE_ebayes_results_table[1:6, ]
DGELUSC_CCLE_top6_hits

head(DGELUSC_CCLE_voomed$E)
DGELUSC_CCLE_top6_hits_df = DGELUSC_CCLE_voomed$E[row.names(DGELUSC_CCLE_top6_hits), ]
DGELUSC_CCLE_top6_hits

DGELUSC_CCLE_top7_hits = DGELUSC_CCLE_ebayes_results_table[1:7, ]
DGELUSC_CCLE_top7_hits

dim(DGELUSC_CCLE_ebayes_results_table) 

par(mfrow = c(2,3))
for (i in 1:nrow(DGELUSC_CCLE_top6_hits_df)) {
  boxplot(row.name(DGELUSC_CCLE_top6_hits)[i]
          }

boxplot(list(quartile1 = DGELUSC_CCLE_top6_hits[DGELUSC_CCLE_top6_hits$GencodeID[i], quartile1_patients], quartile2 = DGELUSC_CCLE_top6_hits[DGELUSC_CCLE_top6_hits$GencodeID[i], quartile2_patients], quartile3 = DGELUSC_CCLE_top6_hits[DGELUSC_CCLE_top6_hits$GencodeID[i], quartile3_patients], quartile4 = DGELUSC_CCLE_top6_hits[DGELUSC_CCLE_top6_hits$GencodeID[i], quartile4_patients]), ylab = "Normalized Expression", main = ovyrs_group_top6_hits$GeneName[i], names = c("Q1", "Q2", "Q3", "Q4"))

```


Plot top hits in heatmap (padj < 0.1)

```{r, echo = FALSE, fig.width = 10, fig.height = 6, warning = FALSE, message = FALSE}
# gene heatmap ----

DGELUSC_CCLE_heatmap_genes = subset(DGELUSC_CCLE_ebayes_results_table, adj.P.Val <= 0.1)
DGELUSC_CCLE_heatmap_df = DGELUSC_CCLE_voomed$E[row.names(DGELUSC_CCLE_heatmap_genes), ]

DGELUSC_CCLE_heatmap_df_transform = as.matrix(t(scale(t(DGELUSC_CCLE_heatmap_df), scale = TRUE, center = TRUE)))
head(DGELUSC_CCLE_heatmap_df_transform)
dim(DGELUSC_CCLE_heatmap_df_transform)
#row.names(DGELUSC_CCLE_heatmap_df_transform) = row.names(DGELUSC_CCLE_heatmap_genes)

library(ggbiplot)
# range(clinical_meta_ovyrs$ovyrs)
DGELUSC_CCLE_col = colorRamp2(c(12, 32, 52), c("white", "lightblue", "darkblue"))

?HeatmapAnnotation
#column_anno = HeatmapAnnotation(
    #ovyrs = clinical_meta_ovyrs$ovyrs,
    #col = list("ovyrs" = DGELUSC_CCLE_col)
)

# create heatmap and draw
gene_heatmap = Heatmap(DGELUSC_CCLE_heatmap_df_transform, cluster_columns = TRUE, cluster_rows = TRUE, row_title = "Gene Expression", column_title = "CCLE Cell lines", name = "Z-score", heatmap_legend_param = list(title_position = "topcenter", legend_direction = "vertical"), show_row_names = FALSE, row_names_gp = gpar(fontsize = 8), show_column_names = TRUE)
# draw(gene_heatmap, heatmap_legend_side = "right", annotation_legend_side = "right", merge_legend = TRUE)
gene_heatmap


head(DGELUSC_CCLE_heatmap_df_transform)


```

\newpage
### Gene set enrichment analysis (Hallmarks of Cancer, Reactome, KEGG)

```{r, echo = FALSE, warning = FALSE, message = FALSE}
## gsea
ovyrs_cont_gsea_ranks = -log10(ovyrs_cont_ebayes_results_table$P.Value)
names(ovyrs_cont_gsea_ranks) = ovyrs_cont_ebayes_results_table$GeneName
```

Hallmarks

```{r, echo = FALSE, warning = FALSE, message = FALSE}
ovyrs_cont_hallmarks = as.data.frame(fgsea(hallmarks, ovyrs_cont_gsea_ranks, minSize = 15, maxSize = 500, scoreType = "pos"))
ovyrs_cont_hallmarks_top10 = head(ovyrs_cont_hallmarks[order(ovyrs_cont_hallmarks$padj), ], 10)
plotGseaTable(hallmarks[ovyrs_cont_hallmarks_top10$pathway], ovyrs_cont_gsea_ranks, ovyrs_cont_hallmarks)
```

\newpage
Reactome

```{r, echo = FALSE, warning = FALSE, message = FALSE}
ovyrs_cont_reactome = as.data.frame(fgsea(reactome, ovyrs_cont_gsea_ranks, minSize = 15, maxSize = 500, scoreType = "pos"))
ovyrs_cont_reactome_top10 = head(ovyrs_cont_reactome[order(ovyrs_cont_reactome$padj), ], 10)
plotGseaTable(reactome[ovyrs_cont_reactome_top10$pathway], ovyrs_cont_gsea_ranks, ovyrs_cont_reactome)
```

\newpage
KEGG

```{r, echo = FALSE, warning = FALSE, message = FALSE}
ovyrs_cont_kegg = as.data.frame(fgsea(kegg, ovyrs_cont_gsea_ranks, minSize = 15, maxSize = 500, scoreType = "pos"))
ovyrs_cont_kegg_top10 = head(ovyrs_cont_kegg[order(ovyrs_cont_kegg$padj), ], 10)
plotGseaTable(kegg[ovyrs_cont_kegg_top10$pathway], ovyrs_cont_gsea_ranks, ovyrs_cont_kegg)
```

Saving limma topTable (file name is endpoint_covariate1+covariate2+..._date) to 

`ovyrs_cont_birthDecade+agedx+cohort_2020-10-23.csv` 
 
standard errors to 
 
`ovyrs_cont_ebayes_SE_2020-10-23.csv` 

GSEA Hallmark results to 

`ovyrs_cont_hallmarks_2020-10-23.csv`

and GSEA Reactome results to 

`ovyrGroup_reactome_2020-10-23.csv`

and GSEA KEGG results to 

`ovyrGroup_kegg_2020-10-23.csv`



```{r, echo = FALSE}
ovyrs_cont_hallmarks$leadingEdge = unlist(lapply(ovyrs_cont_hallmarks$leadingEdge, function(x) {
    unlisted_strip = str_replace_all(x, fixed("\""), "")
    unlisted_strip = str_replace_all(unlisted_strip, "c\\(|\\)|\\\n", "")
    paste(unlisted_strip, collapse = ",")
}))

ovyrs_cont_reactome$leadingEdge = unlist(lapply(ovyrs_cont_reactome$leadingEdge, function(x) {
    unlisted_strip = str_replace_all(x, fixed("\""), "")
    unlisted_strip = str_replace_all(unlisted_strip, "c\\(|\\)|\\\n", "")
    paste(unlisted_strip, collapse = ",")
}))

ovyrs_cont_kegg$leadingEdge = unlist(lapply(ovyrs_cont_kegg$leadingEdge, function(x) {
    unlisted_strip = str_replace_all(x, fixed("\""), "")
    unlisted_strip = str_replace_all(unlisted_strip, "c\\(|\\)|\\\n", "")
    paste(unlisted_strip, collapse = ",")
}))

```

```{r, echo = FALSE, warning = FALSE, message = FALSE}
write.table(ovyrs_cont_ebayes_results_table, "work/ovyrs_cont_birthDecade+agedx+cohort_2020-10-23.csv", row.names = FALSE, sep = ",", quote = T)
ovyrs_cont_se = sqrt(ovyrs_cont_dge_efit$s2.post) * ovyrs_cont_dge_efit$stdev.unscaled
write.table(ovyrs_cont_se, "work/ovyrs_cont_ebayes_SE_2020-10-23.csv", row.names = TRUE, sep = ",", quote = T)

write.table(ovyrs_cont_hallmarks, "work/ovyrs_cont_hallmarks_2020-10-23.csv", row.names = F, sep = ",", quote = T)
write.table(ovyrs_cont_reactome, "work/ovyrs_cont_reactome_2020-10-23.csv", row.names = F, sep = ",", quote = T)
write.table(ovyrs_cont_kegg, "work/ovyrs_cont_kegg_2020-10-23.csv", row.names = F, sep = ",", quote = T)
```







#Old DESeq code
```{r}
CN <- round(CCLEtranscript, digits=0)

df <- CN


### Check the distribution
par(mfrow = c(2,2), mar = c(4,4,1,1))
print( apply(df, 2, function(x) quantile(as.numeric(x))) )
log1 <- apply(df, 2, function(x) {hist(log2(x), breaks = 100)})


idExp <- apply(df, 1, function(x) any(x > 16))
dfExp <- df[idExp, ]
print(dim(dfExp))

library('DESeq2')

table2 <- data.frame(name = colnames(dfExp),
                     condition = CCLE_LUSC_meta$wilkerson_classifier)
dds <- DESeqDataSetFromMatrix(dfExp, 
                              colData=table2, design= ~ condition)
dds <- DESeq(dds)

norCounts <- counts(dds, normalized=TRUE)

res <- results(dds)

head(res)
print(res)
writexl(res, "")

## extract the genes with adjP < 0.0001
resSig = res[ which(res$padj < 0.01), ]
dim(resSig)

sigNorData <- norCounts[which(res$padj < 0.01),]

library(RColorBrewer)
library(gplots)
library(ggplot2)
hmcol =  colorRampPalette(brewer.pal(9, "GnBu"))(100)


heatMapDF_nor <- t( apply(sigNorData, 1, function(x){(x-mean(x))/sd(x)}) )

colnames(heatMapDF_nor) <- c('Young_1','Young_2','Young_3','Young_4',
                             'Aged_1','Aged_2','Aged_3','Aged_4',
                             'AgedNPC86_1','AgedNPC86_2','AgedNPC86_3','AgedNPC86_4')

p <- heatmap.2(heatMapDF_nor, col = hmcol, trace="none", margin=c(10, 10),labRow = F)

heatmap.2(heatMapDF_nor, Rowv=T, Colv=FALSE, dendrogram ="row", #scale ="row", col = col,
          density.info="none", trace="none", margins = c(10,10))
pdf('heatmap.pdf')
p
# to find how many with fold change greater than 1 or less than -1

length(which(resSig$log2FoldChange > 1))
length(which(resSig$log2FoldChange < -1))


res_plot      <- data.frame( res )
res_plot$col  <- 'gray40'
res_plot$col[res_plot$log2FoldChange > 1 & res_plot$padj < 0.01] <- 'red'
res_plot$col[res_plot$log2FoldChange < -1 & res_plot$padj < 0.01] <- 'cornflowerblue'

plot(res_plot$log2FoldChange,
     -log10(res_plot$padj),
     col = res_plot$col, pch = 19, xlab = 'log2(fold change)',
     ylab = '-log10(p-adj)', 
     las = 1 # the direction of ylab will be change
)


```