---
title: "Subset_CCLERNASeq"
author: "AL"
date: "2023-08-27"
output: html_document
---


#Load libraries
```{r setup, include=FALSE}
#Load libraries and setwd
library(here)
library(readxl)
library(stringr)
library(ggplot2)
library(dplyr)
library(tidyr)
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 60), tidy = TRUE)
set.seed(2022)

getwd()
```

#Load files
```{r pressure, echo=FALSE}

#RNAseq_reads (RSEM) Rows: cell lines (Broad IDs) and Columns: genes (HGNC symbol and Ensembl ID)
CCLE_RNAseq_reads = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/CCLE_RNAseq_reads.csv", header=TRUE, stringsAsFactors = FALSE, row.names = 1 )

sample_meta = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/sample_info.csv", header = TRUE)

CCLEmutations = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/CCLE_mutations.csv", header = TRUE)


CCLEcelllines_classifiedbyWilkerson = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/LUSCclassifers_WilkersonTable2.csv",header=TRUE, stringsAsFactors = FALSE)


```


#add TP53 mutations to sample_meta
```{r}
#CCLE mutations has 1235466 x 32 with 1771 unique DepMap_IDs, use columns called DepMap_ID, Hugo_Symbol, Protein_Change and subset with only tp53 mutations

CCLEmutations_subset_tp53 = CCLEmutations

CCLEmutations_subset_tp53 = CCLEmutations_subset_tp53 %>%
  subset(Hugo_Symbol == "TP53") %>%
  subset(Protein_Change != "")
#total of 1119 unique DepMap_IDs with TP53 protein mutations
colnames(CCLEmutations_subset_tp53)[colnames(CCLEmutations_subset_tp53) == "Protein_Change"] <- "p53"
CCLEmutations_subset_tp53_df = CCLEmutations_subset_tp53[,c("DepMap_ID", "p53")]
  
sample_meta = left_join(sample_meta, CCLEmutations_subset_tp53_df)
cellIDS = sample_meta[,1:3]
#sample_meta has all cell lines with p53 protein mutation information added

#subtype using cellosauric_NCIt ... for Lung squamous cell carcinoma, then subset for available RNAseq data
sample_meta_LUSC = subset(sample_meta, sample_meta$Cellosaurus_NCIt_disease == "Lung squamous cell carcinoma") #37 cell lines 



```


#make CCLE dataframe
```{r}
RNASEQ = CCLE_RNAseq_reads #insert which RNAseq csv you want to use, DepMap_ID is used for identifying cell lines

cellsOfInterest = sample_meta_LUSC$DepMap_ID #make vector of cells of interest by Depmap ID#

#Move forward with only matched cells - note making a new table using subset function but with all gene transcripts from original
MatchedCellsTable = subset(RNASEQ, row.names(RNASEQ) %in% cellsOfInterest) #28 cell lines 


#get genes names from RNAseq have the HUGO name followed by the ENSEMBL information, need to only keep HUGO name
CCLEtranscript_colnames = colnames(MatchedCellsTable)
CCLEtranscript_colnames = str_replace(CCLEtranscript_colnames, "\\.\\..*", "")
colnames(MatchedCellsTable) = CCLEtranscript_colnames

#currently has DepMap_IDs as colnames need to change to cell names
MatchedCellsTable_names = data.frame(rownames(MatchedCellsTable))
colnames(MatchedCellsTable_names) = "DepMap_ID"
MatchedCellsTable_names =  left_join(MatchedCellsTable_names, cellIDS)
MatchedCellsTable_names = distinct(MatchedCellsTable_names)
rownames(MatchedCellsTable) = MatchedCellsTable_names$stripped_cell_line_name


#write.csv(MatchedCellsTable, file="~/Project_LUSC/RScripts_LUSC/Exports/CCLE_RNAseq_reads_LUSC.csv")

sample_meta_LUSC = subset(sample_meta_LUSC, sample_meta_LUSC$stripped_cell_line_name %in% MatchedCellsTable_names$stripped_cell_line_name)
sample_meta_LUSC <- sample_meta_LUSC[!duplicated(sample_meta_LUSC$DepMap_ID), ]


#write.csv(sample_meta_LUSC, file="~/Project_LUSC/RScripts_LUSC/Exports/sample_meta_LUSC.csv")


```



Goal output are two files subsetted to LUSC
1) sample_meta_LUSC
2) CCLE_RNAseq_reads_LUSC

