---
title: "CCLE_LUSC"
author: "AL"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
#Load libraries and setwd
library(here)
library(readxl)
library(stringr)
library(ConsensusClusterPlus)
library(ComplexHeatmap)
library(circlize)
library(enrichR)
library(immunedeconv)
library(ggplot2)
library(iClusterPlus)
library(dplyr)
library(tidyr)
library(reshape2)
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 60), tidy = TRUE)
set.seed(2022)

getwd()

```

## Load files
Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.

```{r pressure, echo=FALSE}
#do once - did not repeat because loading data takes too long
#read in RNAseq info takes a long time to load
CCLEtranscript = read.csv("~/Documents/Rstudio/CCLE_RNAseq_transcripts.csv", stringsAsFactors = FALSE)
CCLEtranscript = read.csv("~/Documents/Project_LUSC/RScripts_LUSC/Resources/CCLEDownloads/CCLE_RNAseq_transcripts.csv", stringsAsFactors = FALSE)

#pull out only TP63 transcripts
CCLEtranscript[1:5,1:5]
#saw only uses depmap ID need to match to samples
dim(CCLEtranscript)
      
CCLEcolumn = colnames(CCLEtranscript) 
#every column name is a geneID
CCLEcolumnNumbers_forTP63 = grep("TP63", CCLEcolumn)
CCLEcolumnNumbers_forTP63


#add depmap # to columns wanted
CCLEcolumnNumbers_forTP63
CCLEcolumnNumbers_forTP63 <- c(1, CCLEcolumnNumbers)
CCLEcolumnNumbers_forTP63
length(CCLEcolumnNumbers_forTP63)
#add column 1 for depmap id#

#group TP63 identifiers 

TP63transcriptsisoform_identifiers = read.csv("~/Documents/Rstudio/CCLE_sampleinfo/Transcripts_p63Ntermtype.csv", stringsAsFactors = FALSE)
TP63transcriptsisoform_identifiers

#separate into TA v deltaN vectors
TAp63_isoforms = TP63transcriptsisoform_identifiers$Nterm %in% "TAp63"
TAp63_isoforms = subset(TP63transcriptsisoform_identifiers, TP63transcriptsisoform_identifiers$Nterm %in% "TAp63")
TAp63_isoforms
dim(TAp63_isoforms)
TAp63_isoforms = TAp63_isoforms[,1]
TAp63_isoforms

deltaNp63_isoforms = subset(TP63transcriptsisoform_identifiers, TP63transcriptsisoform_identifiers$Nterm %in% "deltaNp63")
deltaNp63_isoforms
dim(deltaNp63_isoforms)
deltaNp63_isoforms = deltaNp63_isoforms[,1]
deltaNp63_isoforms


#sample info
sample_meta = read.csv("~/Documents/Rstudio/LUSC/scripts/Scripts/CCLEfiles/sample_info.csv", stringsAsFactors = FALSE)
dim(sample_meta)
sample_meta[1:5,1:5]


#pull out sample_meta file make mini df with depmapID and strippedcellname only
sample_meta_names = sample_meta[,c(1,3)]
sample_meta_names



#replace CCLEtranscript depmap id to stripped cell line name for downstream labeling
CCLEtranscript_CellsRelabel = CCLEtranscript

# doesn't work because the lengths of the df are different  --- try to use dplyr
#CCLEtranscript_CellsRelabel[CCLEtranscript_CellsRelabel$X == sample_meta$DepMap_ID] <- sample_meta$stripped_cell_line_name

#all names match from CCLE to sample meta
compare = table(CCLEtranscript_CellsRelabel$X %in% sample_meta$DepMap_ID)
compare
compare = table(sample_meta$DepMap_ID %in% CCLEtranscript_CellsRelabel$X)
compare
# The sample meta has 434 that don't match the CCLE

CCLEtranscript_CellsRelabel[1:5,1:5]
#given up on relabeling too large file, will relabel at end

```

```{r pressure, echo=FALSE}
#RUN WHOLE BLOCK

#identify specific cells that matched

MatchedCells = cellsOfInterest %in% CCLEtranscript$X
MatchedCells
table(MatchedCells)

notMatchedCells = !cellsOfInterest %in% CCLEtranscript$X
notMatchedCells
table(MatchedCells)

#identify which cell lines were not matched
cellsOfInterest[notMatchedCells]
cellsOfInterest

#Move forward with only matched cells - note making a new table using subset function but with all gene transcripts from original
MatchedCells = subset(CCLEtranscript, CCLEtranscript$X %in% cellsOfInterest)
dim(MatchedCells)


#make new table with only interested rows and columns
CCLEtable = MatchedCells[,CCLEcolumnNumbers_forTP63]
head(CCLEtable)
dim(CCLEtable)
CCLEtable

```




```{r pressure, echo=FALSE}
#RUN WHOLE BLOCK TO REORGANIZE CELL LINE NAMES TO FIRST COLUMN AND REMOVE DEPMAP ID#

#check column names are cell type, then each TP63 isoform identifier
CCLEtable_colnames = colnames(CCLEtable)
CCLEtable_colnames

compare = TP63transcriptsisoform_identifiers$Transcript.ID %in% CCLEtable_colnames
compare
#all match



#rename depmapID to cell line names
CCLEtable = left_join(CCLEtable, sample_meta_names, by = c("X" = "DepMap_ID"))
CCLEtable

#move cell line name first and remove X
CCLEtable = CCLEtable %>%
  select(stripped_cell_line_name, everything())

CCLEtable_cells = CCLEtable[,-2]
CCLEtable_cells

```
```{r pressure, echo=FALSE} 
#DON'T DO THIS
#sum expression levels based on TAp63 or deltaNp63 isoforms
#won't allow to transpose then join .... will try to subset then combine into a data frame
#LUSCtable_colnames = colnames(LUSCtable_cells)
# = data.frame(LUSCtable_colnames)
#LUSCtable_colnames
#dim(LUSCtable_colnames)
#colnames(LUSCtable_colnames)[1]="TP63"


#join to label p63 isoform
#LUSCtable_colnames = left_join(LUSCtable_colnames, TP63transcriptsisoform_identifiers, by = c("TP63" = "Transcript.ID"))
#LUSCtable_colnames

#remove original name and then put back into table as types
#LUSCtable_cells_isoforms = LUSCtable_cells
#LUSCtable_cells_isoforms


#colnames(LUSCtable_cells_isoforms) = LUSCtable_colnames$Nterm

#LUSCtable_cells_isoforms
#have table with titles only TAp63 v deltaNp63 now need to sum and graph


#sort_order = sort(colnames(LUSCtable_cells_isoforms))
#LUSCtable_cells_isoforms = LUSCtable_cells_isoforms[,sort_order]
#LUSCtable_cells_isoforms
#this was really dumb - next time you can just order the isoform file by TAp63 and deltaNp63 or combine the two vectors you created for both and used that to order the columns so that it can retain the transcript information. This way you also lost your row names
```

```{r pressure, echo=FALSE}
# run if you want bar plots of each cell line for TAp63 or deltaNp63 - need to edit titles and axis labels


#sum expression levels based on TAp63 or deltaNp63 isoforms
#won't allow to transpose then join .... will try to subset then combine into a data frame
CCLEtable
colnames(CCLEtable)
names(CCLEtable)[1] = "Cell_Line"
names(CCLEtable)[2] = "DepMap_ID"
CCLEtable

length(deltaNp63_isoforms)
#8
length(TAp63_isoforms)
#6

sortorder = c("Cell_Line","DepMap_ID",TAp63_isoforms,deltaNp63_isoforms)
sortorder

CCLEtable_isoforms = CCLEtable[,sortorder]
CCLEtable_isoforms
dim(CCLEtable_isoforms)
# in correct order note that TAp63 are columns 3:8, and deltaNp63 are 9:16

TAp63_sum = rowSums(CCLEtable_isoforms[,c(3:8)], na.rm=TRUE)
deltaNp63_sum = rowSums(CCLEtable_isoforms[,c(9:16)], na.rm = TRUE)

CCLEtable_isoforms_avgs = data.frame(CCLEtable_isoforms[,1],
                                          TAp63_sum,
                                         deltaNp63_sum )

CCLEtable_isoforms_avgs
write.csv(CCLEtable_isoforms_avgs,file="EXPORT_LUAD_p63isoformsCCLEdata.csv")









#Also save this for future analysis 
CCLEtable_LUAD = CCLEtable
CCLEtable_isoforms_avgs_LUAD = CCLEtable_isoforms_avgs

#make barplots for each TP63 isoform
barplot(CCLEtable_isoforms_avgs$TAp63_sum, names = CCLEtable_isoforms_avgs$CCLEtable_isoforms...1., las = 2, main = "CCLE LUAD Cell lines - TAp63 Expression", ylab = "summed expected counts")



barplot(CCLEtable_isoforms_avgs$deltaNp63_sum, names = CCLEtable_isoforms_avgs$CCLEtable_isoforms...1., las = 2, main = "CCLE LUAD Cell lines - deltaNp63 Expression", ylab = "summed expected counts")


```

```{r pressure, echo=FALSE}
#NEED TO EDIT TITLES

#create a stacked bar chart
CCLEtable_isoforms_avgs
newcolnamesfor_t = CCLEtable_isoforms_avgs[,1]
newcolnamesfor_t

t_CCLEtable_isoforms_avgs = t(CCLEtable_isoforms_avgs)
t_CCLEtable_isoforms_avgs[,1:5]
t_CCLEtable_isoforms_avgs = t_CCLEtable_isoforms_avgs [2:3,]
t_CCLEtable_isoforms_avgs[,1:5]
colnames(t_CCLEtable_isoforms_avgs) = newcolnamesfor_t
t_CCLEtable_isoforms_avgs

barplot(t_CCLEtable_isoforms_avgs, las = 2, main = "CCLE LUAD Cell lines - TP63 Isoform Expression", ylab = "summed expected counts", legend.text = c("TAp63", "deltaNp63"))



#create a stacked barchart normalized to 100%, https://statisticsglobe.com/scale-bars-of-stacked-barplot-to-100-percent-in-r
#example didn't work
#barplot(prop.table(t_LUSCtable_cells_isoforms_avgs, ))

#need to transform the data manually first into percentages using base R https://r-graph-gallery.com/211-basic-grouped-or-stacked-barplot.html

colnames(t_CCLEtable_isoforms_avgs)[1]="key"
t_CCLEtable_isoforms_avgs

CCLEtable_isoforms_avgs
colnames(CCLEtable_isoforms_avgs) = c("cells","TAp63","deltaNp63")
CCLEtable_isoforms_avgs
#check for zeros in both columns
CCLEtable_isoforms_avgs[20,]
#yes zeros in both
CCLEtable_isoforms_avgs = CCLEtable_isoforms_avgs[-20,]
CCLEtable_isoforms_avgs


CCLEtable_isoforms_avgs_long = CCLEtable_isoforms_avgs %>%
    gather('TAp63', 'deltaNp63', key = 'cell type', value ='isoform')
CCLEtable_isoforms_avgs_long
#moved into a longer version of a dataframe
colnames(CCLEtable_isoforms_avgs_long) = c("Cell_Line", "TP63_isoform", "Summed_Counts") 
CCLEtable_isoforms_avgs_long
#make stacked plot using ggplot
stacked_summed = ggplot(CCLEtable_isoforms_avgs_long, aes(fill=TP63_isoform, y=Summed_Counts, x=Cell_Line)) + 
    geom_bar(position="stack", stat="identity") + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
stacked_summed


data_percentage = CCLEtable_isoforms_avgs_long
data_percentage = data_percentage %>%
    group_by(cells) %>%
    mutate(percent = 100*isoform/sum(isoform))
data_percentage
colnames(data_percentage) = c("Cell_Line", "TP63_isoform", "Summed_Counts", "percent") 
#concatenated another column for percentages

stacked = ggplot(data_percentage, aes(x = Cell_Line, y = percent, fill = TP63_isoform)) +
  geom_bar(position = "fill", stat = "identity") + 
  scale_y_continuous(labels = scales::percent_format()) + 
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
stacked
#stacked bar plots per cell type



```

# cells of interest from CCLE
```{r pressure, echo=FALSE}
cellsOfInterest
#currently looking at LUAD cells of interest
length(cellsOfInterest)

#Wilkerson subtype
#download expression full  RNAseq data genes for gene level data instead of transcript variants we were using for analysis above
CCLE_ExpFull_RNAseq = read.csv("~/Documents/Rstudio/LUSC/CCLE_expression_full.csv", stringsAsFactors = FALSE, check.names=FALSE, header=TRUE, row.names=1)
CCLE_ExpFull_RNAseq[1:5,1:5]
dim(CCLE_ExpFull_RNAseq)
#  1406 19222 and genes are columns and depmap ID/cells are rows under column name X


compare = cellsOfInterest %in% row.names(CCLE_ExpFull_RNAseq)
table(compare)
#false 3 and true 38

MatchedCells = cellsOfInterest %in% row.names(CCLE_ExpFull_RNAseq)
MatchedCells
table(MatchedCells)
#confirmed 3 don't match, move forward with only matched cells
CCLE_ExpFull_RNAseq_subset = subset(CCLE_ExpFull_RNAseq, row.names(CCLE_ExpFull_RNAseq) %in% cellsOfInterest)
dim(CCLE_ExpFull_RNAseq_subset)
#see 38 rows - okay, but still 19222 columns need to trim to wilkerson gene set


#upload gene signatures from wilkerson files
wilkerson_centroid = read.csv("~/Documents/Rstudio/LUSC/wilkerson/centroids_all_cohorts_all_common_genes.csv", row.names = 1)
head(wilkerson_centroid)
Wilkerson_genes = row.names(wilkerson_centroid)
Wilkerson_genes
length(Wilkerson_genes)

#get genes in CCLE_ExpFull_RNAseq
CCLE_colnames = colnames(CCLE_ExpFull_RNAseq)
length(CCLE_colnames)
CCLE_colnames[1:5]
#removed all characters after . in names
CCLE_colnames = gsub(" .*","",CCLE_colnames)
CCLE_colnames[1:5]

#reapply colnames
CCLE_ExpFull_RNAseq_subset_NoNums=CCLE_ExpFull_RNAseq_subset
colnames(CCLE_ExpFull_RNAseq_subset_NoNums) = CCLE_colnames
head(CCLE_ExpFull_RNAseq_subset_NoNums)



compare = Wilkerson_genes %in% CCLE_colnames
head(compare)
table(compare)
compare_genes = Wilkerson_genes[compare]

compare2 = CCLE_colnames %in%  Wilkerson_genes
table(compare2)

#genes # matched 8421, checked genes on HUGO and saw some gene names are outdated
Wilkerson_not_matching_genes = Wilkerson_genes[!compare]
length(Wilkerson_not_matching_genes) #940 matches check before
#CCLE_colnames[grep("^KRT.*", CCLE_colnames, ignore.case = FALSE)]


CCLE_ExpFull_RNAseq_subset_NoNums[1:5,1:5]
CCLE_ExpFull_RNAseq_subset_NoNums[,1]
dim(CCLE_ExpFull_RNAseq_subset_NoNums)

#cannot do - transpose first
CCLE_Wilkerson <- CCLE_ExpFull_RNAseq_subset_NoNums[,compare_genes]
dim(CCLE_Wilkerson)
CCLE_Wilkerson[1:5,1:5]

t_CCLE_Wilkerson = t(CCLE_Wilkerson)
t_CCLE_Wilkerson[1:5,1:5]
dim(t_CCLE_Wilkerson)
#subset -- genes are rows and samples are columns

```



#Generate long df of Mapping_table from Eric Welsh with symbols and alias for use
```{r}
#read in excel table for gene matching from Eric Welsh
Mapping_table = read.csv("~/Documents/Rstudio/LUSC/gencode_v19_merged_gene_transcript_annotation.csv", stringsAsFactors = FALSE, header = TRUE)
head(Mapping_table)


#pull out symbols and alias column from mapping_table
head(Mapping_table)
Mapping_table_symbol_alias = select(Mapping_table, symbol, alias)
head(Mapping_table_symbol_alias)

#remove all rows with empty values in alias
Mapping_table_symbol_alias = Mapping_table_symbol_alias[!(Mapping_table_symbol_alias$alias == ""),]
head(Mapping_table_symbol_alias)
dim(Mapping_table_symbol_alias)

#what is the total amount of new columns you will need (how many spaces are there in the largest alias plus 1 for the extra alias)
ncols <- max(stringr::str_count(Mapping_table_symbol_alias$alias, " ")) + 1
ncols #31 columns after splitting

#make column names
colmn = paste ("alias", 1:ncols)
colmn


#split multiple names under alias separated by space
Mapping_table_symbol_alias = separate(Mapping_table_symbol_alias, col = alias, sep = " ", into = colmn)
head(Mapping_table_symbol_alias)

#make into a long table using gather from tidyr
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias %>%
  gather(all_of(colmn), key = 'amount', value ='Alias')

head(Mapping_table_symbol_alias_long)
Mapping_table_symbol_alias_long
dim(Mapping_table_symbol_alias_long) #696322 x 3

#check if NA's were included, check to see if other aliases were included because only see alias1
length(which(Mapping_table_symbol_alias_long$amount=="alias 10")) 
length(which(Mapping_table_symbol_alias_long$amount=="alias 30")) 
#for every alias# there is 22462 guessing all the NAs were included need to remove
length(which(Mapping_table_symbol_alias_long$amount=="NA")) #this says 0, may be it is blank?
#didn't work    length(which(Mapping_table_symbol_alias_long$amount==is.na)) 
table(sapply(Mapping_table_symbol_alias_long, is.na)) #643286 is TRUE, 1445680 is false
#remove empty cells
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias_long[-which(is.na(Mapping_table_symbol_alias_long$Alias)),]
dim(Mapping_table_symbol_alias_long) #53036 x 3

#order by gene name
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias_long[order(Mapping_table_symbol_alias_long$symbol),]
head(Mapping_table_symbol_alias_long)
length(which(Mapping_table_symbol_alias_long$amount=="alias 10")) #74
length(which(Mapping_table_symbol_alias_long$amount=="alias 30")) #1

write.csv(Mapping_table_symbol_alias_long, file="~/Documents/Rstudio/LUSC/Gene_Mapping_Alias_Long.csv")


# don't want to know how many alias there is per gene can remove column 2
Mapping_table_symbol_alias_long = select(Mapping_table_symbol_alias_long, -amount)
head(Mapping_table_symbol_alias_long)
#finally have mapping table with column 1 being the symbol and column two being the alias

```

#clean up genes comparisons
```{r}
head(Wilkerson_not_matching_genes)

compare = select(Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$symbol)
table(compare) #579 FALSE, and 361 TRUE


compare = Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$Alias
table(compare) #382 FALSE, and 558 TRUE


compare1 = subset(Wilkerson_not_matching_genes, Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$symbol)
length(compare1) #361

compare2 = subset(Wilkerson_not_matching_genes, Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$Alias)
length(compare2) #558


compare_both = compare1 %in% compare2
table(compare_both) #352 FALSE, 9 TRUE

compare_both = subset(compare1, compare1 %in% compare2)
compare_both


compare_both = compare2 %in% compare1
table(compare_both) #549 FALSE, 9 TRUE

compare_both = subset(Wilkerson_not_matching_genes, compare1 %in% compare2)
compare_both


#unique from symbols
unique_compare1 = subset(compare1, !(compare1 %in% compare2))
unique_compare1
#combine for a total covered list
pot_target_gene_names = c(unique_compare1, compare2)
length(pot_target_gene_names) #910
#check for duplicates
table(duplicated(pot_target_gene_names)) #910 FALSE

unmatched_wilkerson = subset(Wilkerson_not_matching_genes, !(Wilkerson_not_matching_genes %in% pot_target_gene_names))
length(unmatched_wilkerson) #30


#need to change names for Wilkerson_not_matching genes 

#also need to check gene names for RNAseq 


```


#Change Wilkerson alias to symbol names
```{r}
head(wilkerson_centroid)
colnames(wilkerson_centroid)
dim(wilkerson_centroid)
#issue that column 1 has the gene names and don't know how to target it - row names is the gene alias - read in data again and not put row.names = TRUE
wilkerson_centroid_mappedgenes = wilkerson_centroid

head(wilkerson_centroid_mappedgenes)
#row.names(wilkerson_centroid_mappedgenes)[1] = "Ashley" ---this changed the first row to name Ashley
  

#first check to see if wilkerson gene centroid names match CCLE RNAseq, 
#if not then check alias table and see if alias match and change to gene symbol name
head(Mapping_table_symbol_alias_long) #alias table
head(CCLE_colnames) #gene names from the CCLE RNAseq 

for (i in 1:nrow(wilkerson_centroid_mappedgenes)) {
  print(i)
  current_gene = row.names(wilkerson_centroid_mappedgenes)[i]
  # need if statement here; does the current_gene match CCLE?
  if (!current_gene %in% CCLE_colnames) {
    # does not match, need to get aliases for this gene symbol
    if (current_gene %in% Mapping_table_symbol_alias_long$Alias) {
      # gets the alias to gene symbol mapping
      current_gene_aliases = subset(Mapping_table_symbol_alias_long, Alias == current_gene)
      for (j in 1:nrow(current_gene_aliases)) {
        # check if the mapping/new gene symbol is already in Wilkerson
        if (!current_gene_aliases$symbol[j] %in% row.names(CCLE_colnames)) {
        # assigns the mapping back into the wilkerson_centroid_mappedgenes data frame
        row.names(wilkerson_centroid_mappedgenes)[i] = current_gene_aliases$symbol
      }
      }
    }
  }
}

```


#Try For Loop
```{r}

#try with selected genes - arbitrarily picked line 1
current_gene = Wilkerson_not_matching_genes


#only 53 matched, looking at alias column, different alias are separated by spaces--- write another for loop and it will be nested


#make sure you use length for this vector and nrow for df
#for(i in 1:length(Wilkerson_not_matching_genes)){
  print(Wilkerson_not_matching_genes[i])
  #if stop here then will only print the 940 gene names
  current_gene = Wilkerson_not_matching_genes[i]
  #every iteration new variable current_gene will be the gene of interest at the moment
  
  #current_gene is hopefully the alias and we want to change it to a symbol
#  }

  
  #Paul's method - one loop to scan every row, the see if there a space, then do nested loop to split string or use str_match function to see if
```