---
title: "Wilkerson_Centroids"
output: html_document
date: "2023-08-28"
---

```{r setup, include=FALSE}
#Load libraries and setwd
library(here)
library(readxl)
library(stringr)
library(dplyr)
library(tidyr)
library(reshape2)
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 60), tidy = TRUE)
set.seed(2022)

getwd()
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

#reads data:Wilkerson, CCLE_RNAseq and sample info
```{r cars}

CCLE_RNAseq_reads_LUSC = read.csv(file="~/Project_LUSC/RScripts_LUSC/Exports/CCLE_RNAseq_reads_LUSC.csv", header = TRUE, row.names = 1)
#subsetted LUSC cell lines with RNAseq genes with X as the colname for stripped_cell_line and the colnames being genes

wilkerson_centroid = read.csv("~/Project_LUSC/RScripts_LUSC/Exports/centroids_all_cohorts_all_common_genes.csv", row.names = 1)


sample_meta_LUSC = read.csv(file = "~/Project_LUSC/RScripts_LUSC/Exports/sample_meta_LUSC.csv", header = TRUE)
```







#matching genes between wilkerson centroid and CCLE RNAseq data
```{r}
CCLERNAseq_colnames = colnames(CCLE_RNAseq_reads_LUSC) #54357

Wilkerson_genes = row.names(wilkerson_centroid)

#Goal compare gene names from Wilkerson centroid to gene names from RNASeq
compare = Wilkerson_genes %in% CCLERNAseq_colnames
table(compare) #False 956 and True 8405

#identify unmatched genes names to change later
Wilkerson_not_matching_genes = Wilkerson_genes[!compare]

#need to try to find a way to change the 956 unmatched genes - emailed Eric Welsh and tried to use his gene mapping table

```
#Generate long df of Mapping_table from Eric Welsh with symbols and alias for use
```{r}
#read in excel table for gene matching from Eric Welsh
Mapping_table = read.csv(file = "~/Project_LUSC/RScripts_LUSC/Resources/gencode_v19_merged_gene_transcript_annotation.csv", stringsAsFactors = FALSE, header = TRUE)

Mapping_table_symbol_alias = select(Mapping_table, symbol, alias)
Mapping_table_symbol_alias = Mapping_table_symbol_alias[!(Mapping_table_symbol_alias$alias == ""),] #remove all rows with empty values in alias


#what is the total amount of new columns you will need (how many spaces are there in the largest alias plus 1 for the extra alias)
ncols <- max(stringr::str_count(Mapping_table_symbol_alias$alias, " ")) + 1  #31 columns after splitting
colmn = paste ("alias", 1:ncols) #make column names
Mapping_table_symbol_alias = separate(Mapping_table_symbol_alias, col = alias, sep = " ", into = colmn) #split multiple names under alias separated by space


#make into a long table using gather from tidyr
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias %>%
  gather(all_of(colmn), key = 'amount', value ='Alias')
#696322 x 3


#table(sapply(Mapping_table_symbol_alias_long, is.na)) #643286 is TRUE, 1445680 is false
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias_long[-which(is.na(Mapping_table_symbol_alias_long$Alias)),] #remove empty cells
Mapping_table_symbol_alias_long = Mapping_table_symbol_alias_long[order(Mapping_table_symbol_alias_long$symbol),] #order by gene name 
#length(which(Mapping_table_symbol_alias_long$amount=="alias 10")) #74


#write.csv(Mapping_table_symbol_alias_long, file="~/Documents/Rstudio/LUSC/Gene_Mapping_Alias_Long.csv")

Mapping_table_symbol_alias_long = select(Mapping_table_symbol_alias_long, -amount) #no need for alias counts, now have mapping table with column 1 being the symbol and column 2 being the alias

```


#check comparisons between wilkerson and alias table
```{r}

compare = Wilkerson_not_matching_genes %in% Mapping_table_symbol_alias_long$symbol 
table(compare) #374 true (likely can't change these) and 582 false
list = Wilkerson_not_matching_genes[compare]
compare = list %in% CCLERNAseq_colnames 
table(compare) #all 374 not there -- as expected


#hard to know if the correct ones will be changed will be <582 that might be changed definitely the 374 won't be changed
compare = CCLERNAseq_colnames %in% Mapping_table_symbol_alias_long$Alias
table(compare) #398 FALSE, and 558 TRUE
list = CCLERNAseq_colnames[compare]
length(list) #2141 are there

```


#Change Wilkerson alias to symbol names
```{r}

wilkerson_centroid_mappedgenes = wilkerson_centroid

for (i in 1:nrow(wilkerson_centroid_mappedgenes)) {
  print(i)
  current_gene = row.names(wilkerson_centroid_mappedgenes)[i]
  # need if statement here; does the current_gene match CCLE?
  if (!current_gene %in% CCLERNAseq_colnames) {
    # does not match, need to get aliases for this gene symbol
    if (current_gene %in% Mapping_table_symbol_alias_long$Alias) {
      # gets the alias to gene symbol mapping
      current_gene_aliases = subset(Mapping_table_symbol_alias_long, Alias == current_gene)
      for (j in 1:nrow(current_gene_aliases)) {
        # check if the mapping/new gene symbol is already in Wilkerson
        if (!current_gene_aliases$symbol[j] %in% row.names(CCLERNAseq_colnames)) {
        # assigns the mapping back into the wilkerson_centroid_mappedgenes data frame
        row.names(wilkerson_centroid_mappedgenes)[i] = current_gene_aliases$symbol
      }
      }
    }
  }
}
#stopped at 702 because duplicate row.names not allowed


wilkerson_centroid_mappedgenes = wilkerson_centroid[-c(702,856, 920, 1121, 2875, 3070, 3268, 4546, 5134, 5221, 5452, 5784, 6616, 7300),]
#duplicate at 856, 920, 1121, 2875, 3070, 3268, 4546, 5134, 5221, 5452, 5784, 6616, 7300
#now there are 9347 genes the original wilkerson centroid had 9361


for (i in (1:nrow(wilkerson_centroid_mappedgenes))) {
  print(i)
  current_gene = row.names(wilkerson_centroid_mappedgenes)[i]
  # need if statement here; does the current_gene match CCLE?
  if (!current_gene %in% CCLERNAseq_colnames) {
    # does not match, need to get aliases for this gene symbol
    if (current_gene %in% Mapping_table_symbol_alias_long$Alias) {
      # gets the alias to gene symbol mapping
      current_gene_aliases = subset(Mapping_table_symbol_alias_long, Alias == current_gene)
      for (j in 1:nrow(current_gene_aliases)) {
        # check if the mapping/new gene symbol is already in Wilkerson
        if (!current_gene_aliases$symbol[j] %in% row.names(CCLERNAseq_colnames)) {
        # assigns the mapping back into the wilkerson_centroid_mappedgenes data frame
        row.names(wilkerson_centroid_mappedgenes)[i] = current_gene_aliases$symbol
      }
      }
    }
  }
}


#write.csv(wilkerson_centroid_mappedgenes, file = "~/Documents/CCLE_LUSC/Exports/wilkerson_centroid_mappedgenes_edited9347.csv")

```



#run Wilkerson
```{r}
#Goal compare gene names from Wilkerson centroid to gene names from RNASeq
sample_meta_LUSCsubtyped = sample_meta_LUSC

t_CCLE_RNAseq_reads_LUSC = t(CCLE_RNAseq_reads_LUSC)
dim(t_CCLE_RNAseq_reads_LUSC) #28 cells

compare = row.names(wilkerson_centroid_mappedgenes) %in% CCLERNAseq_colnames #8899 true and 448 false compared to 8405 and 956 from before

genes_for_wilkerson = intersect(row.names(wilkerson_centroid_mappedgenes), row.names(t_CCLE_RNAseq_reads_LUSC))
length(genes_for_wilkerson) #8899

gene_wilkerson_corrs = cor(t_CCLE_RNAseq_reads_LUSC[genes_for_wilkerson,], wilkerson_centroid_mappedgenes[genes_for_wilkerson, ])
dim(gene_wilkerson_corrs) # 28 x 4 Col.  

sample_meta_LUSCsubtyped$wilkerson_classifier = apply(gene_wilkerson_corrs, 1, function(x) {names(x)[which.max(x)]}) #7 basal, 16 classical, 5 primitive, no secretory

write.csv(sample_meta_LUSCsubtyped, file = "~/Project_LUSC/RScripts_LUSC/Exports/sample_meta_LUSCsubtyped.csv")
```


Goal: export single files
1) sample_meta_LUSCsubtyped.csv with predicted Wilkerson subtypes











#ISIS run wilkerson
```{r}
#Goal compare gene names from Wilkerson centroid to gene names from RNASeq
compare = row.names(wilkerson_centroid_mappedgenes) %in% colnames(MatchedCellsTable) 
table(compare) #False 448 (before 956) and True 8899 (before 8405) sooo 956 of the Wilkerson genes don't match what is in the CCLE RNASeq

genes_for_wilkerson = intersect(row.names(wilkerson_centroid_mappedgenes), row.names(counts))
length(genes_for_wilkerson) #8899

gene_wilkerson_corrs = cor(counts[genes_for_wilkerson,], wilkerson_centroid_mappedgenes[genes_for_wilkerson, ])
dim(gene_wilkerson_corrs) # 28 x 4 Col.  

SampleInfoSLC$wilkerson_classifier = apply(gene_wilkerson_corrs, 1, function(x) {names(x)[which.max(x)]}) 
head(SampleInfoSLC)

write.csv(SampleInfoSLC, file = "~/Project_LUSC/RScripts_LUSC/Exports/SampleInfoSLC_wilkerson_classifier.csv")
```