---
title: "Wilkerson_Centroids"
output: html_document
date: "2023-08-28"
---

```{r setup, include=FALSE}
#Load libraries and setwd
library(here)
library(readxl)
library(stringr)
library(ConsensusClusterPlus)
library(ComplexHeatmap)
library(circlize)
library(enrichR)
library(immunedeconv)
library(ggplot2)
library(iClusterPlus)
library(dplyr)
library(tidyr)
library(reshape2)
library(limma)
library(edgeR)
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 60), tidy = TRUE)
set.seed(2022)

getwd()
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

#reads data:Wilkerson, CCLE_RNAseq and sample info
```{r cars}

wilkerson_centroid_mappedgenes <- read.csv("~/SquamousLung_MultiOmics/wilkerson_centroid_mappedgenes_edited9347.csv", row.names=1)
head(wilkerson_centroid_mappedgenes)

MatchedCellsTable <- read.csv("~/SquamousLung_MultiOmics/MatchedCellsTable_CCLERNASeq_reads.csv", row.names=1)
dim(MatchedCellsTable) # 28 x 54357 #row names are cells and colnames are genes
head(MatchedCellsTable)
hist(apply(MatchedCellsTable,1, max))
#counts should be in the thousands to millions - SO GOOD

## Sample info
sample_info <- read.csv("~/SquamousLung_MultiOmics/sample_info.csv")
SampleInfoSLC = merge(MatchedCellsTable[,c(1,2)], sample_info, by.x= 0, by.y = "stripped_cell_line_name")
SampleInfoSLC = SampleInfoSLC[,-c(2,3)]
```

#CCLE_RNAseq_reads data
```{r}
counts = t(MatchedCellsTable) #transform so genes are row names and cells are column
dim(counts) #54357 28

counts_na = counts
counts_na[counts_na == 0] = NA
boxplot(log2(counts_na)) #median looks about the same across- likely normalized
```

#run Wilkerson
```{r}
#Goal compare gene names from Wilkerson centroid to gene names from RNASeq
compare = row.names(wilkerson_centroid_mappedgenes) %in% colnames(MatchedCellsTable) 
table(compare) #False 448 (before 956) and True 8899 (before 8405) sooo 956 of the Wilkerson genes don't match what is in the CCLE RNASeq

genes_for_wilkerson = intersect(row.names(wilkerson_centroid_mappedgenes), row.names(counts))
length(genes_for_wilkerson) #8899

gene_wilkerson_corrs = cor(counts[genes_for_wilkerson,], wilkerson_centroid_mappedgenes[genes_for_wilkerson, ])
head(gene_wilkerson_corrs) # 28 x 4 Col.  

SampleInfoSLC$wilkerson_classifier = apply(gene_wilkerson_corrs, 1, function(x) { names(x)[which.max(x)]}) 
head(SampleInfoSLC)

write.csv(SampleInfoSLC, file = "SampleInfoSLC_wilkerson_classifier.csv")
```
